---
title: "Genotype quality control with plinkQC"
author: "Hannah Meyer"
date: "`r Sys.Date()`"
output: 
    pdf_document:
        fig_caption: yes
        toc: true
        toc_depth: 2
vignette: >
  %\VignetteIndexEntry{GenotypeQC_with_plinkQC}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
devtools::load_all()
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Introduction
Genotyping arrays allow to measure a samples' genotype at thousands of markers. 
Subsequent studies such as genome-wide association studies rely on high quality
genotypes which can be ensure for thorough quality control protocols. plinkQC
follows the quality control proposed by Anderson and colleagues @Anderson2010.
It assumes the genotype have already been determined from the original probe
intensity data of the genotype array and is available in [plink format](https://www.cog-genomics.org/plink/1.9/input#bed). 

plinkQC depends on the [plink](https://www.cog-genomics.org/plink/1.9/), which
has to be manually installed prior to the usage of plinkQC. 

plinkQC offers functions for 'per-sample' and 'per-marker' QC. As advocated in 
@Anderson2010, functions are implemented in a way that the 'per-sampleâ€™ QC
is conducted before the  'per-marker'to maximize the number of markers
remaining in the study. This approach prevents markers from being erroneously
removed because of a subset of poorly genotyped individuals. Contrarily,
samples may be falsely excluded based on a poorly genotyped subset of markers.
However, both QCs may be conducted independently.

The following paragraphs will briefly describe the purpose of each per-sample
and per-marker QC step (for more details, see @Anderson2010).
This general description is followed by a step-by-step genotype QC of an example
dataset using plinkQC.

## Per-sample QC
The per-sample QC of genotypes consists of four steps: (i) identification of
samples with discordant sex information, (ii) identification of samples with
outlying missing genotype and/or heterozygosity rates, (iii) identification of
related samples and (iv) identification of samples of divergent ancestry.

The identification of samples with discordant sex information helps to detect
sample mix-ups and samples with very poor genotyping rates. For each sample, the
homozygosity rates across all X-chromosomal genetic markers are computed and
compared with the expected rates: <0.2 for females and >0.8 for males. For
samples where the assigned sex (PEDSEX in the .fam file) contradicts the sex
inferred from the homozygosity rates, it should be checked that the sex was
correctly recorded (genotyping often occurs at different locations as
phenotyping and misrecording might occur). Samples with discordant sex
information that is not accounted for should be removed from the study.

The identification of samples with outlying missing genotype and/or
heterozygosity rates helps to detect samples with poor DNA quality and/or
concentration that should be excluded from the study.

Depending on the future use of the genotypes, it might required to remove any
related samples from the study. Related individuals can be identified by their
proporting of shared alleles at the genotyped markers (identity by
descend, IBD). Standardly, samples with second-degree relatedness of higher will
be excluded. 

The identification of samples of divergent ancestry can be achieved by
combining the genotypes of the study population with genotypes of a reference
dataset consisting of samples from known ethnicities (for instance samples from
the Hapmap or 1000 genomes study @Hapmap, @1000Genomes). Principal component
analysis on this combined genotype panle can be used to detect population
structure down to the level of the reference dataset (for Hapmap and 1000
Genomes, this down to large-scale continental ancestry).

## Per-marker QC
The per-marker QC consists of (i) identifying markers with an large missing
genotype rate, (ii) identifying markers showing a significant deviation from
Hardy-Weinberg equilibrium (HWE) and (iii) the removal markers with low minor
allele frequency (MAF)/minor allele count.

Markers with excessive missingness rate are removed as they are considered 
unreliable. Markers with strong deviation from HWE might be indicative of
genotyping or genotype-calling errors. Markers with low minor allele count are 
often removed as the actual genotype calling (via the calling algorithm) is very
difficult due to the small sizes of the heterozygote and rare-homozygote
clusters.


The following vignette shows the usage of the individual and marker QC functions
and depicts summaries of the QCs via its overview functions. 

# Running plink
```{r set parameters}
package.dir <- find.package('plinkQC')
qcdir <- "~/RPackagesDevel/plinkQC/inst/extdata"
#qcdir <- file.path(package.dir, 'extdata')
alg <- 'data'
path2plink <- "~/bin"
```

```{r run qc}

```

# Per-sample QC
The per-sample QC of genotypes is implemented in the following functions:

  1. identification of samples with discordant sex information: `check_sex`
  2. identification of samples with outlying missing
    genotype and/or heterozygosity rates:
    `check_heterozygosity_and_missingness`
  3. identification of related samples: `check_relatedness`
  4. identification of samples of divergent ancestry: `check_ancestry`.

These checks can be carried out individually or run jointly via `perSampleQC`,
where one can specify which check to run. Per default all checks are conducted.

In addition to running each check, `perSampleQC` writes a list of all fail
sample IDs to the qcdir. These IDs will be removed in the computation of the
`perMarkerQC`. If the list is not present, `perMarkerQC` will send a message
about conducting the QC on the entire dataset.

In the following, the per-sample QC of the results of `run_QC.sh` is depicted.
Within `perSampleQC`, one simply specifies the directory where the data is
stored and the prefix of the plink files (i.e. prefix.bim, prefix.bed,
prefix.fam). In addition, the names of the files containing information about 
the reference population and the merged dataset used in `check_ancestry` have to be
provided: refSamplesFile, refColorsFile and prefixMergedDataset. Per default, 
all QC checks will be conducted. 

## QC steps
`perSampleQC` displays the results of the QC steps in a multi-panel plot.

```{r sample QC,  fig.height=12, fig.width=9}
fail_samples <- perSampleQC(qcdir=qcdir, alg=alg,
                            refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",
                                                 sep=""), 
                            refColorsFile=paste(qcdir, "/HapMap_PopColors.txt",
                                                 sep=""),
                            prefixMergedDataset="data.HapMapIII",
                            interactive=TRUE, verbose=TRUE)

```

\newpage
## Overview
`overviewPerSampleQC` depicts overview plots of QC failures and the intersection
of QC failures with ancestry exclusion.
```{r overview sample QC}
overview_samples <- overviewPerSampleQC(fail_samples, interactive=TRUE)
```

# Per-marker QC
The per-marker QC of genotypes is implemented in the following functions:

  1. identification of markers with high missingness rates:
    `check_snp_missingnes`
  2. identification of markers showing significant deviation from HWE:
    `check_hwe`
  3. identification of markers with a very low minor allele frequency:
    `check_maf`
These checks can be carried out individually or run jointly via `perMarkerQC`,
where one can specify which check to run. Per default all checks are conducted.

## QC steps
`perMarkerQC` displays the results of the QC step in a multi-panel plot.
```{r marker QC, fig.height=9, fig.width=6, fig.show='hold'}
par(mfrow=c(2,1), las=1)
fail_markers <- perMarkerQC(qcdir=qcdir, alg=alg, path2plink=path2plink,
                            verbose=TRUE, interactive=TRUE,
                            showPlinkOutput=FALSE)
```

\newpage
## Overview
`overviewPerMarkerQC` depicts overview plots of QC failures.
```{r overview marker QC}
overview_marker <- overviewPerMarkerQC(fail_markers, interactive=TRUE)
```

