\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Processing HapMap III reference data for ancestry estimation},
            pdfauthor={Hannah Meyer},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\newenvironment{Shaded}{}{}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.56,0.13,0.00}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.53,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.73,0.40,0.53}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.73,0.13,0.13}{\textit{#1}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.02,0.16,0.49}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.10,0.09,0.49}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.40,0.40,0.40}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.74,0.48,0.00}{#1}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.49,0.56,0.16}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{#1}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Processing HapMap III reference data for ancestry estimation}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Hannah Meyer}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2018-10-24}


\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\section{Introduction}\label{introduction}

Genotype quality control for genetic association studies often includes
the need for selecting samples of the same ethnic background. To
identify individuals of divergent ancestry based on genotypes, the
genotypes of the study population can be combined with genotypes of a
reference dataset consisting of individuals from known ethnicities.
Principal component analysis (PCA) on this combined genotype panel can
then be used to detect population structure down to the level of the
reference dataset.

The following vignette shows the processing steps required to use
samples of the HapMap study {[}1{]}{[}2{]}{[}3{]} as a reference
dataset. Using this reference, population structure down to large-scale
continental ancestry can be detected. A step-by-step instruction on how
to conduct this analysis is described in this
\href{https://hannahvmeyer.github.io/plinkQC/articles/AncestryCheck.html}{vignette}.

\section{Workflow}\label{workflow}

\subsection{Set-up}\label{set-up}

We will first set up some bash variables and create directories needed;
storing the names and directories of the reference will make it easy to
use updated versions of the reference in the future. Is is also useful
to keep the PLINK log-files for future reference. In order to keep the
data directory tidy, we'll create a directory for the log files and move
them to the log directory here after each analysis step.

\begin{Shaded}
\begin{Highlighting}[]
\VariableTok{refdir=}\StringTok{'~/reference'}
\FunctionTok{mkdir}\NormalTok{ -r }\VariableTok{$qcdir}\NormalTok{/plink_log}
\end{Highlighting}
\end{Shaded}

\subsection{Download and convert Hapmap phase III
data}\label{download-and-convert-hapmap-phase-iii-data}

Hapmap phase 3 data (HapMapIII) is available in
\href{https://www.cog-genomics.org/plink/1.9/input\#ped}{PLINK text
format} at ncbi. The following code chunk downloads and unzips the data.

\begin{Shaded}
\begin{Highlighting}[]
\BuiltInTok{cd} \VariableTok{$refdir}

\VariableTok{ftp=}\NormalTok{ftp://ftp.ncbi.nlm.nih.gov/hapmap/genotypes/2009-01_phaseIII/plink_format/}
\VariableTok{prefix=}\NormalTok{hapmap3_r2_b36_fwd.consensus.qc.poly}

\FunctionTok{wget} \VariableTok{$ftp}\NormalTok{/}\VariableTok{$prefix}\NormalTok{.map.bz2}
\FunctionTok{bunzip2} \VariableTok{$prefix}\NormalTok{.map.bz2}

\FunctionTok{wget} \VariableTok{$ftp}\NormalTok{/}\VariableTok{$prefix}\NormalTok{.ped.bz2}
\FunctionTok{bunzip2} \VariableTok{$prefix}\NormalTok{.per.bz2}
\end{Highlighting}
\end{Shaded}

We then convert the
\href{https://www.cog-genomics.org/plink/1.9/input\#ped}{PLINK text
format} to the standardly used
\href{https://www.cog-genomics.org/plink/1.9/inputbped}{PLINK binary
format}.

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{plink}\NormalTok{ --file }\VariableTok{$refdir}\NormalTok{/}\VariableTok{$prefix}\NormalTok{ \textbackslash{}}
\NormalTok{      --make-bed \textbackslash{}}
\NormalTok{      --out }\VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36}
\FunctionTok{mv} \VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36.log }\VariableTok{$refdir}\NormalTok{/log}
\end{Highlighting}
\end{Shaded}

\subsection{Update annotation}\label{update-annotation}

The genome build of HapMap III data is NCBI36. Currently most datasets
are updated to CGRCh37 or CGRCh38. In order to update the HapMap III
data to the desired build, we use the UCSC
\href{https://genome.ucsc.edu/cgi-bin/hgLiftOver}{liftOver} tool. The
liftOver tool takes information in a format similar to the
\href{https://www.cog-genomics.org/plink/1.9/formats\#bim}{PLINK .bim}
format, the
\href{https://genome.ucsc.edu/FAQ/FAQformat.html\#format1}{UCSC bed
format} and a
\href{http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/}{liftover
chain}, containing the mapping information between the old genome
(target) and new genome (query). It returns the updated annotation
(newFile) and a file with unmappable variants (unMapped):

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{liftOver}\NormalTok{ oldFile liftover.chain newFile unMapped}
\end{Highlighting}
\end{Shaded}

We first need to download the liftOver tool from
\url{https://genome.ucsc.edu/cgi-bin/hgLiftOver} and the appropriate
liftover chain from
\url{http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/}). We then
convert the
\href{https://www.cog-genomics.org/plink/1.9/formats\#bim}{PLINK .bim}
format, to the zero-based
\href{https://genome.ucsc.edu/FAQ/FAQformat.html\#format1}{UCSC bed}
format.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{awk} \StringTok{'\{print "chr" $1, $4 -1, $4, $2 \}'} \VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36.bim }\KeywordTok{|} \KeywordTok{\textbackslash{}}
    \KeywordTok{\{} \FunctionTok{sed} \StringTok{'s/chr23/chrX/'}\KeywordTok{;} \FunctionTok{sed} \StringTok{'s/chr24/chrY/'}\KeywordTok{\}} \OperatorTok{>} \KeywordTok{\textbackslash{}}
    \VariableTok{$refdir}\ExtensionTok{/HapMapIII_NCBI36.tolift}
\end{Highlighting}
\end{Shaded}

We use the liftOver tool and the UCSC bed formated annotation file
together with the appropriate chain file to do the lift over.

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{liftOver} \VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36.tolift }\VariableTok{$refdir}\NormalTok{/hg18ToHg19.over.chain \textbackslash{}}
    \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37 }\VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36.unMapped}
\end{Highlighting}
\end{Shaded}

After successful liftover, we will be able to extract i) the variants
that were mappable from the old to the new genome and ii) their updated
positions

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# ectract mapped variants}
\FunctionTok{awk} \StringTok{'\{print $4\}'} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37 }\OperatorTok{>} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37.snps}
\CommentTok{# ectract updated positions}
\FunctionTok{awk} \StringTok{'\{print $4, $3\}'} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37 }\OperatorTok{>} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37.pos}
\end{Highlighting}
\end{Shaded}

\subsection{Update the reference data}\label{update-the-reference-data}

We can now use PLINK to extract the mappable variants from the old build
and update their position. After these steps, the HapMap III dataset can
be used for infering study ancestry as described in the corresponding
\href{https://hannahvmeyer.github.io/plinkQC/articles/AncestryCheck.html}{vignette}.

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{plink}\NormalTok{ --bfile }\VariableTok{$refdir}\NormalTok{/HapMapIII_NCBI36}
    \ExtensionTok{--extract} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37.snps \textbackslash{}}
\NormalTok{    --update-map }\VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37.pos \textbackslash{}}
\NormalTok{    --make-bed \textbackslash{}}
\NormalTok{    --out }\VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37}
\FunctionTok{mv} \VariableTok{$refdir}\NormalTok{/HapMapIII_CGRCh37.log }\VariableTok{$refdir}\NormalTok{/log}
\end{Highlighting}
\end{Shaded}

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\hypertarget{ref-HapMap2005}{}
1. The International HapMap Consortium. A haplotype map of the human
genome. Nature. 2005;437: 1299--320.
doi:\href{https://doi.org/10.1038/nature04226}{10.1038/nature04226}

\hypertarget{ref-HapMap2007}{}
2. The International HapMap Consortium. A second generation human
haplotype map of over 3.1 million SNPs. Nature. 2007;449: 851.
doi:\href{https://doi.org/10.1038/nature06258}{10.1038/nature06258}

\hypertarget{ref-HapMap2010}{}
3. The International HapMap Consortium. Integrating common and rare
genetic variation in diverse human populations. Nature. 2010;467.
doi:\href{https://doi.org/10.1038/nature09298}{10.1038/nature09298}


\end{document}
