---
title: "Genotype quality control with plinkQC"
author: "Hannah Meyer"
date: "`r Sys.Date()`"
output: 
    pdf_document:
        fig_caption: yes
        toc: true
        toc_depth: 2
<<<<<<< HEAD
bibliography: references.bib
csl: plos-genetics.csl
=======
>>>>>>> 1df06936c00b6131a09818f88b954b5158edd35f
vignette: >
  %\VignetteIndexEntry{GenotypeQC_with_plinkQC}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
library(plinkQC)
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Introduction
Genotyping arrays allow to measure a samples' genotype at thousands of markers. 
Subsequent studies such as genome-wide association studies rely on high quality
genotypes which can be ensure for thorough quality control protocols. plinkQC
follows the quality control proposed by Anderson and colleagues @Anderson2010.
It assumes the genotype have already been determined from the original probe
intensity data of the genotype array and is available in [plink format](https://www.cog-genomics.org/plink/1.9/input#bed). 

plinkQC depends on the [plink](https://www.cog-genomics.org/plink/1.9/), which
has to be manually installed prior to the usage of plinkQC. 

plinkQC offers functions for 'per-sample' and 'per-marker' QC. As advocated in 
@Anderson2010, functions are implemented in a way that the 'per-sampleâ€™ QC
is conducted before the  'per-marker'to maximize the number of markers
remaining in the study. This approach prevents markers from being erroneously
removed because of a subset of poorly genotyped individuals. Contrarily,
samples may be falsely excluded based on a poorly genotyped subset of markers.
However, both QCs may be conducted independently.

The following paragraphs will briefly describe the purpose of each per-sample
and per-marker QC step (for more details, see @Anderson2010).
This general description is followed by a step-by-step genotype QC of an example
dataset using plinkQC.

## Per-sample QC
The per-sample QC of genotypes consists of four steps: (i) identification of
samples with discordant sex information, (ii) identification of samples with
outlying missing genotype and/or heterozygosity rates, (iii) identification of
related samples and (iv) identification of samples of divergent ancestry.

The identification of samples with discordant sex information helps to detect
sample mix-ups and samples with very poor genotyping rates. For each sample, the
homozygosity rates across all X-chromosomal genetic markers are computed and
compared with the expected rates: <0.2 for females and >0.8 for males. For
samples where the assigned sex (PEDSEX in the .fam file) contradicts the sex
inferred from the homozygosity rates, it should be checked that the sex was
correctly recorded (genotyping often occurs at different locations as
phenotyping and misrecording might occur). Samples with discordant sex
information that is not accounted for should be removed from the study.

The identification of samples with outlying missing genotype and/or
heterozygosity rates helps to detect samples with poor DNA quality and/or
concentration that should be excluded from the study.

Depending on the future use of the genotypes, it might required to remove any
related samples from the study. Related individuals can be identified by their
proporting of shared alleles at the genotyped markers (identity by
descend, IBD). Standardly, samples with second-degree relatedness of higher will
be excluded. 

The identification of samples of divergent ancestry can be achieved by
combining the genotypes of the study population with genotypes of a reference
dataset consisting of samples from known ethnicities (for instance samples from
the Hapmap or 1000 genomes study @Hapmap, @1000Genomes). Principal component
analysis on this combined genotype panle can be used to detect population
structure down to the level of the reference dataset (for Hapmap and 1000
Genomes, this down to large-scale continental ancestry).

## Per-marker QC
The per-marker QC consists of (i) identifying markers with an large missing
genotype rate, (ii) identifying markers showing a significant deviation from
Hardy-Weinberg equilibrium (HWE) and (iii) the removal markers with low minor
allele frequency (MAF)/minor allele count.

Markers with excessive missingness rate are removed as they are considered 
unreliable. Markers with strong deviation from HWE might be indicative of
genotyping or genotype-calling errors. Markers with low minor allele count are 
often removed as the actual genotype calling (via the calling algorithm) is very
difficult due to the small sizes of the heterozygote and rare-homozygote
clusters.


The following vignette shows the usage of the individual and marker QC functions
and depicts summaries of the QCs via its overview functions. 

# Running plink
```{r set parameters}
package.dir <- find.package('plinkQC')
qcdir <- file.path(package.dir, 'extdata')
alg <- 'data'
path2plink <- "~/bin"
```

```{r run qc}

```

# Per-sample QC
The per-sample QC of genotypes is implemented in the following functions:

  1. identification of samples with discordant sex information: `check_sex`
  2. identification of samples with outlying missing
    genotype and/or heterozygosity rates:
    `check_heterozygosity_and_missingness`
  3. identification of related samples: `check_relatedness`
  4. identification of samples of divergent ancestry: `check_ancestry`.

These checks can be carried out individually or run jointly via `perSampleQC`,
where one can specify which check to run. Per default all checks are conducted.

In addition to running each check, `perSampleQC` writes a list of all fail
sample IDs to the qcdir. These IDs will be removed in the computation of the
`perMarkerQC`. If the list is not present, `perMarkerQC` will send a message
about conducting the QC on the entire dataset.

In the following, the per-sample QC of the results of `run_QC.sh` is depicted.
Within `perSampleQC`, one simply specifies the directory where the data is
stored and the prefix of the plink files (i.e. prefix.bim, prefix.bed,
prefix.fam). In addition, the names of the files containing information about 
the reference population and the merged dataset used in `check_ancestry` have to be
provided: refSamplesFile, refColorsFile and prefixMergedDataset. Per default, 
all QC checks will be conducted. 

## QC steps
`perSampleQC` displays the results of the QC steps in a multi-panel plot.

```{r sample QC,  fig.height=12, fig.width=9}
fail_samples <- perSampleQC(qcdir=qcdir, alg=alg,
                            refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",
                                                 sep=""), 
                            refColorsFile=paste(qcdir, "/HapMap_PopColors.txt",
                                                 sep=""),
                            prefixMergedDataset="data.HapMapIII",
                            interactive=TRUE, verbose=TRUE)

```

\newpage
## Overview
`overviewPerSampleQC` depicts overview plots of QC failures and the intersection
of QC failures with ancestry exclusion.
```{r overview sample QC}
overview_samples <- overviewPerSampleQC(fail_samples, interactive=TRUE)
```

<<<<<<< HEAD

## Per-marker quality control
`perMarkerQC` applies its checks to data in the specified directory (qcdir),
starting with the specified prefix of the plink files (i.e. prefix.bim,
prefix.bed, prefix.fam). Optionally, the user can specify different thresholds
for the quality control checks and which check to conduct. Per default, all
quality control checks will be conducted. `perMarkerQC` displays the results of
the QC step in a multi-panel plot.
```{r marker QC, eval=FALSE}
fail_markers <- perMarkerQC(qcdir=qcdir, alg=alg, path2plink=path2plink,
                            verbose=TRUE, interactive=TRUE,
                            showPlinkOutput=FALSE)
```
```{r load markerQC, echo=FALSE, fig.height=9, fig.width=6, fig.align='center'}
par(mfrow=c(2,1), las=1)
knitr::include_graphics("markerQC.pdf")
```

`overviewPerMarkerQC` depicts an overview of the marker quality control
failures and their overlaps.
```{r overview marker QC, eval=FALSE}
overview_marker <- overviewPerMarkerQC(fail_markers, interactive=TRUE)
```

```{r load overviewMarkerQC, echo=FALSE}
par(mfrow=c(2,1), las=1)
knitr::include_graphics("overviewMarkerQC.pdf")
```

\newpage
## Create QC-ed dataset
After checking results of the per-individual and per-marker quality control,
individuals and markers that fail the chosen criteria can automatically be
removed from the dataset  with `cleanData`, resulting in the new dataset qcdir/data.clean.bed,qcdir/data.clean.bim, qcdir/data.clean.fam. For
convenience, `cleanData` returns a list of all individuals in the study split
into keep and remove individuals.
```{r clean data, eval=FALSE}
Ids  <- cleanData(qcdir=qcdir, alg=alg, path2plink=path2plink,
                            verbose=TRUE, showPlinkOutput=FALSE)
```

# Step-by-step

## Individuals with discordant sex information
The identification of individuals with discordant sex information helps to detect
sample mix-ups and samples with very poor genotyping rates. For each sample, the
homozygosity rates across all X-chromosomal genetic markers are computed and
compared with the expected rates (typically $<$0.2 for females and $>$0.8 for
males). For samples where the assigned sex (PEDSEX in the .fam file) contradicts
the sex inferred from the homozygosity rates (SNPSEX), it should be checked that
the sex was correctly recorded (genotyping often occurs at different locations
as phenotyping and misrecording might occur). Samples with discordant sex
information that is not accounted for should be removed from the study.
Identifying individuals with discordant sex information is implemented in 
`check_sex`. It finds individuals  whose SNPSEX != PEDSEX. Optionally, an extra
data.frame with sample IDs and sex can be provided to double check if external
and PEDSEX data (often processed at different centers) match. If a mismatch
between PEDSEX and SNPSEX was detected, by SNPSEX == Sex, PEDSEX of these
individuals can optionally be updated. `check_sex` depicts the X-chromosomal
heterozygosity (SNPSEX) of the samples split by their (PEDSEX).

```{r check sex, fig.height=3, fig.width=5, fig.align='center'}
fail_sex <- check_sex(qcdir=qcdir, alg=alg, interactive=TRUE, verbose=TRUE,
                      run.check_sex=FALSE)
```

## Individuals with outlying missing genotype and/or heterozygosity rates
The identification of individuals with outlying missing genotype and/or
heterozygosity rates helps to detect samples with poor DNA quality and/or
concentration that should be excluded from the study. Typically, individuals
with more than 3-7\% of their genotype calles missing are removed. Outlying
heterozygosity rates are judged relative to the overall heterozygosity rates in
the study, and individuals whose rates are more than a few standard deviations
(sd) from the mean heterozygosity rate are removed. A typical quality control
for outlying heterozygoisty rates would remove individuals who are three sd away
from the mean rate.
Identifying related individuals with outlying missing genotype and/or
heterozygosity rates is implemented in `check_het_and_miss`.
It finds individuals that have genotyping and heterozygosity rates that fail the
set thresholds and depicts the results as a scatter plot with the samples'
missingness rates on x-axis and their heterozygosity rates on the y-axis.


```{r check het miss, fig.height=3, fig.width=5, fig.align='center'}
fail_het_imiss <- check_het_and_miss(qcdir=qcdir, alg=alg,
                                     interactive=TRUE,
                                     run.check_het_and_miss=FALSE)
```


## Related individuals
Depending on the future use of the genotypes, it might required to remove any
related individuals from the study. Related individuals can be identified by
their proporting of shared alleles at the genotyped markers (identity by
descend, IBD). Standardly, individuals with second-degree relatedness or higher
will be excluded. 
Identifying related individuals is implemented in `check_relatedness`. It finds
pairs of samples whose proportion of IBD is larger than the specified highIBDTh. 
Subsequently, for pairs of individual that do not have additional relatives in
the dataset, the individual with the greater genotype missingness rate is
selected and returned as the individual failing the relatedness check. For more
complex family structures, the unrelated individuals per family are selected
(e.g. in a parents-offspring trio, the offspring will be marked as fail, while 
the parents will be kept in the analysis).

```{r check related, fig.height=3, fig.width=5, fig.align='center'}
exclude_relatedness <- check_relatedness(qcdir=qcdir, alg=alg, interactive=TRUE,
                                         run.check_relatedness=FALSE)
```


## Individuals of divergent ancestry
The identification of individuals of divergent ancestry can be achieved by
combining the genotypes of the study population with genotypes of a reference
dataset consisting of individuals from known ethnicities (for instance
individuals from the Hapmap or 1000 genomes study [@HapMap2005,@HapMap2007,
@HapMap2010,@a1000Genomes2015,@b1000Genomes2015]). Principal component
analysis on this combined genotype panel can be used to detect population
structure down to the level of the reference dataset (for Hapmap and 1000
Genomes, this is down to large-scale continental ancestry).
Identifying individuals of divergent ancestry is implemented in
`check_ancestry`. Currently, check ancestry only supports automatic selection of
individuals of European descent. It uses information from principal components 1
and 2 to find the center of the European reference samples. All study samples
whose euclidean distance from the centre falls outside a specified radius are
considered non-European. `check_ancestry` creates a scatter plot of PC1 versus
PC2 color-coded for samples of the reference populations and the study
population.

```{r check ancestry, fig.height=3, fig.width=5, fig.align='center'}
exclude_ancestry <- check_ancestry(qcdir=qcdir, alg=alg,
                            refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",
                                                 sep=""), 
                            refColorsFile=paste(qcdir, "/HapMap_PopColors.txt",
                                                 sep=""),
                            prefixMergedDataset="data.HapMapIII",
                            run.check_ancestry=FALSE,
                            interactive=TRUE)
```

## Markers with excessive missingness rate
Markers with excessive missingness rate are removed as they are considered 
unreliable. Typically, thresholds for marker exclusion based on missingness
range from 1\%-5\%. 
Identifying markers with high missingness rates is implemented in
`snp_missingness`. It calculates the rates of missing genotype calls and
frequency for all variants in the individuals that passed the `perIndividualQC`.

```{r check snp missing, eval=FALSE}
fail_snpmissing <- check_snp_missingness(qcdir=qcdir, alg=alg, interactive=TRUE,
                                         path2plink=path2plink, 
                                         showPlinkOutput=FALSE)
```

```{r load snp missing, echo=FALSE, fig.align='center'}
knitr::include_graphics("snpmissingness.pdf")
```
## Markers with deviation from HWE
Markers with strong deviation from HWE might be indicative of genotyping or
genotype-calling errors. As serious genotyping errors often yield
very low p-values (in the order of $10^-50$), it is recommended to choose a 
reasonably low threshold to avoid filtering too many variants (that might have
slight, non-critical deviations).
Identifying markers with deviation from HWE is implemented in `check_hwe`.
It calculates the observed and expected heterozygote frequencies per SNP in the
individuals that passed the `perIndividualQC` and computes the deviation of the
frequencies from Hardy-Weinberg equilibrium (HWE) by HWE exact test.

```{r check hwe, eval=FALSE}
fail_hwe <- check_hwe(qcdir=qcdir, alg=alg, interactive=TRUE,
                      path2plink=path2plink, showPlinkOutput=FALSE)
```
```{r load hwe, echo=FALSE, fig.align='center'}
knitr::include_graphics("hwe.pdf")
```


## Markers with low minor allele frequency
Markers with low minor allele count are often removed as the actual genotype
calling (via the calling algorithm) is very difficult due to the small sizes of
the heterozygote and rare-homozygote clusters.
Identifying markers with low minor allele count is implemented in `check_maf`.
It calculates the minor allele frequencies for all variants in the individuals
that passed the `perIndividualQC`.

```{r check maf, eval=FALSE}
fail_maf <- check_maf(qcdir=qcdir, alg=alg, interactive=TRUE,
                      path2plink=path2plink, showPlinkOutput=FALSE)
```
```{r load  maf, echo=FALSE, fig.align='center'}
knitr::include_graphics("maf.pdf")
```

# References
