<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Ancestry estimation based on reference samples of known ethnicities • plinkQC</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Ancestry estimation based on reference samples of known ethnicities">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/Genomes1000.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Ancestry estimation based on reference samples of known ethnicities</h1>
                        <h4 class="author">Hannah Meyer</h4>
            
            <h4 class="date">2018-11-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HannahVMeyer/plinkQC/blob/master/vignettes/AncestryCheck.Rmd"><code>vignettes/AncestryCheck.Rmd</code></a></small>
      <div class="hidden name"><code>AncestryCheck.Rmd</code></div>

    </div>

    
    
<div id="ancestry-estimation" class="section level1">
<h1 class="hasAnchor">
<a href="#ancestry-estimation" class="anchor"></a>Ancestry estimation</h1>
<p>The identification of individuals of divergent ancestry can be achieved by combining the genotypes of the study population with genotypes of a reference dataset consisting of individuals from known ethnicities (for instance individuals from the Hapmap or 1000 genomes study <span class="citation">[1]</span>). Principal component analysis (PCA) on this combined genotype panel can then be used to detect population structure down to the level of the reference dataset (for Hapmap and 1000 Genomes, this is down to large-scale continental ancestry).</p>
<p>In the following, the workflow for combining a study dataset with the reference samples, conducting PCA and estimating ancestry is demonstrated. The study dataset consists of 200 individuals and 10,000 genetic markers and is provided with <span class="math inline">\(plinkQC\)</span> in <code>file.path(find.package('plinkQC'),'extdata')</code>.</p>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<div id="download-reference-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-reference-data" class="anchor"></a>Download reference data</h2>
<p>A suitable reference dataset should be downloaded and if necessary, re-formated into PLINK format. Vignettes <a href="https://hannahvmeyer.github.io/plinkQC/articles/HapMap.html">‘Processing HapMap III reference data for ancestry estimation’</a> and <a href="https://hannahvmeyer.github.io/plinkQC/articles/1000Genomes.html">‘Processing 1000Genomes reference data for ancestry estimation’</a>, show the download and processing of the HapMap phase III and 1000Genomes phase III dataset, respectively. In this example, we will use the HammapIII data as the reference dataset.</p>
</div>
<div id="set-up" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set-up</h2>
<p>We will first set up some bash variables and create directories needed; storing the names and directories of the reference and study will make it easy to use updated versions of the reference or new datasets in the future. Is is also useful to keep the PLINK log-files for future reference. In order to keep the data directory tidy, we’ll create a directory for the log files and move them to the log directory here after each analysis step.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">qcdir=</span><span class="st">'~/qcdir'</span>
<span class="va">refdir=</span><span class="st">'~/reference'</span>
<span class="va">name=</span><span class="st">'data'</span>
<span class="va">refname=</span><span class="st">'HapMapIII'</span>

<span class="fu">mkdir</span> -r <span class="va">$qcdir</span>/plink_log</code></pre></div>
</div>
<div id="match-study-genotypes-and-reference-data" class="section level2">
<h2 class="hasAnchor">
<a href="#match-study-genotypes-and-reference-data" class="anchor"></a>Match study genotypes and reference data</h2>
<p>In order to compute joint principal components of the reference and study population, we’ll need to combine the two datasets. The plink –merge function enables this merge, but requires the variants in the datasets to be matching by chromosome, position and alleles. The following sections show how to extract the relevant data from the reference and study dataset and how to filter matching variants.</p>
<div id="prune-study-data" class="section level3">
<h3 class="hasAnchor">
<a href="#prune-study-data" class="anchor"></a>Prune study data</h3>
<p>We will conduct principle component analysis on genetic variants that are pruned for variants in linkage disequlibrium (LD) with an <span class="math inline">\(r^2 &gt;0.2\)</span> in a 50kb window. The LD-pruned dataset is generated below, using plink –indep-pairwise to compute the LD-variants; additionally exclude range is used to remove genomic ranges of known high-LD structure. This file was originally provided by <span class="citation">[6]</span> and is available in <code>file.path(find.package('plinkQC'),'extdata','high-LD-regions.txt')</code>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile  <span class="va">$qcdir</span>/<span class="va">$name</span> \
      --exclude range  <span class="va">$refdir</span>/<span class="va">$highld</span> \
      --indep-pairwise 50 5 0.2 \
      --out <span class="va">$qcdir</span>/<span class="va">$name</span>
<span class="fu">mv</span>  <span class="va">$qcdir</span>/<span class="va">$name</span>.prune.log <span class="va">$qcdir</span>/plink_log/<span class="va">$name</span>.prune

<span class="ex">plink</span> --bfile  <span class="va">$qcdir</span>/<span class="va">$name</span> \
      --extract <span class="va">$qcdir</span>/<span class="va">$name</span>.prune.in \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned
<span class="fu">mv</span>  <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned.log <span class="va">$qcdir</span>/plink_log/<span class="va">$name</span>.pruned</code></pre></div>
</div>
<div id="filter-reference-data-for-the-same-snp-set-as-in-study" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-reference-data-for-the-same-snp-set-as-in-study" class="anchor"></a>Filter reference data for the same SNP set as in study</h3>
<p>We will use the list of pruned variants from the study sample to reduce the reference dataset to the size of the study samples:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile  <span class="va">$refdir</span>/<span class="va">$refname</span> \
      --extract <span class="va">$qcdir</span>/<span class="va">$name</span>.prune.in \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned
<span class="fu">mv</span>  <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned.log <span class="va">$qcdir</span>/plink_log/<span class="va">$refname</span>.pruned</code></pre></div>
</div>
<div id="check-and-correct-chromosome-mismatch" class="section level3">
<h3 class="hasAnchor">
<a href="#check-and-correct-chromosome-mismatch" class="anchor"></a>Check and correct chromosome mismatch</h3>
<p>The following section uses an awk-script to check that the variant IDs of the reference data have the same chromosome ID as the study data. For computing the genetic PC, the annotation is not important, however, merging the files via PLINK will only work for variants with perfectly matching attributes. For simplicity, we update the pruned reference dataset. Note, that sex chromosomes are often encoded differently and might make the matching more difficult. Again, for simplicty and since not crucial to the final task, we will ignore XY-encoded sex chromosomes (via <code>sed -n '/^[XY]/!p'</code>).</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">awk</span> <span class="st">'BEGIN {OFS="\t"} FNR==NR {a[$2]=$1; next} \</span>
<span class="st">    ($2 in a &amp;&amp; a[$2] != $1)  {print a[$2],$2}'</span> \
    <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned.bim <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned.bim <span class="kw">|</span> <span class="kw">\</span>
    <span class="fu">sed</span> -n <span class="st">'/^[XY]/!p'</span> <span class="op">&gt;</span> <span class="va">$qcdir</span>/<span class="va">$refname</span>.toUpdateChr

<span class="ex">plink</span> --bfile <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned \
      --update-chr <span class="va">$qcdir</span>/<span class="va">$refname</span>.toUpdateChr 1 2 \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$refname</span>.updateChr
<span class="fu">mv</span> <span class="va">$qcdir</span>/<span class="va">$refname</span>.updateChr.log <span class="va">$qcdir</span>/plink_log/<span class="va">$refname</span>.updateChr.log</code></pre></div>
</div>
<div id="position-mismatch" class="section level3">
<h3 class="hasAnchor">
<a href="#position-mismatch" class="anchor"></a>Position mismatch</h3>
<p>Similar to the chromosome matching, we use an awk-script to find variants with mis-matching chromosomal positions.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">awk</span> <span class="st">'BEGIN {OFS="\t"} FNR==NR {a[$2]=$4; next} \</span>
<span class="st">    ($2 in a &amp;&amp; a[$2] != $4)  {print a[$2],$2}'</span> \
    <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned.bim <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned.bim <span class="op">&gt;</span> \
    <span class="va">$qcdir</span>/<span class="va">${refname}</span>.toUpdatePos</code></pre></div>
</div>
<div id="possible-allele-flips" class="section level3">
<h3 class="hasAnchor">
<a href="#possible-allele-flips" class="anchor"></a>Possible allele flips</h3>
<p>Unlike chromosomal and base-pair annotation, mismatching allele-annotations will not only prevent the plink –merge, but also mean that it is likely that actually a different genotype was measured. Initially, we can use the following awk-script to check if non-matching allele codes are a simple case of allele flips.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">awk</span> <span class="st">'BEGIN {OFS="\t"} FNR==NR {a[$1$2$4]=$5$6; next} \</span>
<span class="st">    ($1$2$4 in a &amp;&amp; a[$1$2$4] != $5$6 &amp;&amp; a[$1$2$4] != $6$5)  {print $2}'</span> \
    <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned.bim <span class="va">$qcdir</span>/<span class="va">$refname</span>.pruned.bim <span class="op">&gt;</span> \
    <span class="va">$qcdir</span>/<span class="va">$refname</span>.toFlip</code></pre></div>
</div>
<div id="upate-positions-and-flip-alleles" class="section level3">
<h3 class="hasAnchor">
<a href="#upate-positions-and-flip-alleles" class="anchor"></a>Upate positions and flip alleles</h3>
<p>We use plink to update the mismatching positions and possible allele-flips identified above.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile <span class="va">$qcdir</span>/<span class="va">$refname</span>.updateChr \
      --update-map <span class="va">$qcdir</span>/<span class="va">$refname</span>.toUpdatePos 1 2 \
      --flip <span class="va">$qcdir</span>/<span class="va">$refname</span>.toFlip \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$refname</span>.flipped
<span class="fu">mv</span> <span class="va">$qcdir</span>/<span class="va">$refname</span>.flipped.log <span class="va">$qcdir</span>/plink_log/<span class="va">$refname</span>.flipped.log</code></pre></div>
</div>
<div id="remove-mismatches" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-mismatches" class="anchor"></a>Remove mismatches</h3>
<p>Any alleles that do not match after allele flipping, are identified and removed from the reference dataset.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">awk</span> <span class="st">'BEGIN {OFS="\t"} FNR==NR {a[$1$2$4]=$5$6; next} \</span>
<span class="st">    ($1$2$4 in a &amp;&amp; a[$1$2$4] != $5$6 &amp;&amp; a[$1$2$4] != $6$5) {print $2}'</span> \
    <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned.bim <span class="va">$qcdir</span>/<span class="va">$refname</span>.flipped.bim <span class="op">&gt;</span> \
    <span class="va">$qcdir</span>/<span class="va">$refname</span>.mismatch

<span class="ex">plink</span> --bfile <span class="va">$qcdir</span>/<span class="va">$refname</span>.flipped \
      --exclude <span class="va">$qcdir</span>/<span class="va">$refname</span>.mismatch \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$refname</span>.clean
<span class="fu">mv</span> <span class="va">$qcdir</span>/<span class="va">$refname</span>.clean.log <span class="va">$qcdir</span>/plink_log/<span class="va">$refname</span>.clean.log</code></pre></div>
</div>
</div>
<div id="merge-study-genotypes-and-reference-data" class="section level2">
<h2 class="hasAnchor">
<a href="#merge-study-genotypes-and-reference-data" class="anchor"></a>Merge study genotypes and reference data</h2>
<p>The matching study and reference dataset can now be merged into a combined dataset with plink –bmerge. If all steps outlined above were conducted successfully, no mismatch errors should occur.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile <span class="va">$qcdir</span>/<span class="va">$name</span>.pruned  \
      --bmerge <span class="va">$qcdir</span>/<span class="va">$refname</span>.clean.bed <span class="va">$qcdir</span>/<span class="va">$refname</span>.clean.bim \
         <span class="va">$qcdir</span>/<span class="va">$refname</span>.clean.fam  \
      --make-bed \
      --out <span class="va">$qcdir</span>/<span class="va">$name</span>.merge.<span class="va">$refname</span>
<span class="fu">mv</span> <span class="va">$qcdir</span>/<span class="va">$name</span>.merge.<span class="va">$refname</span>.log <span class="va">$qcdir</span>/plink_log</code></pre></div>
</div>
<div id="pca-on-the-merged-data" class="section level2">
<h2 class="hasAnchor">
<a href="#pca-on-the-merged-data" class="anchor"></a>PCA on the merged data</h2>
<p>We can now run principal component analysis on the combined dataset using plink –pca which returns a .eigenvec file with the family and individual ID in columns 1 and 2, followed by the first 20 principal components.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile <span class="va">$qcdir</span>/<span class="va">$name</span>.merge.<span class="va">$refname</span> \
      --pca \
      --out <span class="va">$qcdir</span>/<span class="va">$name</span>.<span class="va">$reference</span>
<span class="fu">mv</span> <span class="va">$qcdir</span>/<span class="va">$name</span>.<span class="va">$reference</span>.log <span class="va">$qcdir</span>/plink_log</code></pre></div>
</div>
<div id="check-ancestry" class="section level2">
<h2 class="hasAnchor">
<a href="#check-ancestry" class="anchor"></a>Check ancestry</h2>
<p>We can use the .eigenvec file to estimate the ancestry of the study samples. Identifying individuals of divergent ancestry is implemented in <code>check_ancestry</code>. Currently, check ancestry only supports automatic selection of individuals of European descent. It uses principal components 1 and 2 to find the center of the known European reference samples. All study samples whose Euclidean distance from the centre falls outside the radius specified by the maximum Euclidean distance of the reference samples multiplied by the chosen <code>europeanTh</code> are considered non-European. <code>check_ancestry</code> shows the result of the ancestry analysis in a scatter plot of PC1 versus PC2 colour-coded for samples of the reference populations and the study population. From within R, run the following command to the ancestry check:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(plinkQC)
indir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="dt">package=</span><span class="st">"plinkQC"</span>)
name &lt;-<span class="st"> 'data'</span>
refname &lt;-<span class="st"> 'HapMapIII'</span>
prefixMergedDataset &lt;-<span class="st"> </span><span class="kw">paste</span>(name, <span class="st">"."</span>, refname, <span class="dt">sep=</span><span class="st">""</span>)

exclude_ancestry &lt;-
<span class="st">    </span><span class="kw"><a href="../reference/evaluate_check_ancestry.html">evaluate_check_ancestry</a></span>(<span class="dt">indir=</span>indir, <span class="dt">name=</span>name,
                            <span class="dt">prefixMergedDataset=</span>prefixMergedDataset,
                            <span class="dt">refSamplesFile=</span><span class="kw">paste</span>(indir, <span class="st">"/HapMap_ID2Pop.txt"</span>,
                                                 <span class="dt">sep=</span><span class="st">""</span>), 
                            <span class="dt">refColorsFile=</span><span class="kw">paste</span>(indir, <span class="st">"/HapMap_PopColors.txt"</span>,
                                                <span class="dt">sep=</span><span class="st">""</span>),
                            <span class="dt">interactive=</span><span class="ot">TRUE</span>)</code></pre></div>
<embed src="checkAncestry.pdf" style="display: block; margin: auto;" type="application/pdf"></embed>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-HapMap2005">
<p>1. The International HapMap Consortium. A haplotype map of the human genome. Nature. 2005;437: 1299–320. doi:<a href="https://doi.org/10.1038/nature04226">10.1038/nature04226</a></p>
</div>
<div id="ref-HapMap2007">
<p>2. The International HapMap Consortium. A second generation human haplotype map of over 3.1 million SNPs. Nature. 2007;449: 851. doi:<a href="https://doi.org/10.1038/nature06258">10.1038/nature06258</a></p>
</div>
<div id="ref-HapMap2010">
<p>3. The International HapMap Consortium. Integrating common and rare genetic variation in diverse human populations. Nature. 2010;467. doi:<a href="https://doi.org/10.1038/nature09298">10.1038/nature09298</a></p>
</div>
<div id="ref-a1000Genomes2015">
<p>4. 1000 Genomes Project Consortium. An integrated map of structural variation in 2,504 human genomes. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a></p>
</div>
<div id="ref-b1000Genomes2015">
<p>5. 1000 Genomes Project Consortium. A global reference for human genetic variation. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15393">10.1038/nature15393</a></p>
</div>
<div id="ref-Anderson2010">
<p>6. Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT. Data quality control in genetic case-control association studies. Nature Protocols. 2010;5: 1564–73. doi:<a href="https://doi.org/10.1038/nprot.2010.116">10.1038/nprot.2010.116</a></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#ancestry-estimation">Ancestry estimation</a></li>
      <li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#download-reference-data">Download reference data</a></li>
      <li><a href="#set-up">Set-up</a></li>
      <li><a href="#match-study-genotypes-and-reference-data">Match study genotypes and reference data</a></li>
      <li><a href="#merge-study-genotypes-and-reference-data">Merge study genotypes and reference data</a></li>
      <li><a href="#pca-on-the-merged-data">PCA on the merged data</a></li>
      <li><a href="#check-ancestry">Check ancestry</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
