            <h4 class="date">2018-11-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HannahVMeyer/plinkQC/blob/master/vignettes/Genomes1000.Rmd"><code>vignettes/Genomes1000.Rmd</code></a></small>
      <div class="hidden name"><code>Genomes1000.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Genotype quality control for genetic association studies often includes the need for selecting samples of the same ethnic background. To identify individuals of divergent ancestry based on genotypes, the genotypes of the study population can be combined with genotypes of a reference dataset consisting of individuals from known ethnicities. Principal component analysis (PCA) on this combined genotype panel can then be used to detect population structure down to the level of the reference dataset.</p>
<p>The following vignette shows the processing steps required to use samples of the 1000 Genomes study <span class="citation">[1]</span>,<span class="citation">[2]</span> as a reference dataset. Using the 1000 Genomes reference, population structure down to large-scale continental ancestry can be detected. A step-by-step instruction on how to conduct this ancestry analysis is described in this <a href="https://hannahvmeyer.github.io/plinkQC/articles/AncestryCheck.html">vignette</a>.</p>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<div id="set-up" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set-up</h2>
<p>We will first set up some bash variables and create directories needed; storing the names and directories of the reference will make it easy to use updated versions of the reference in the future. Is is also useful to keep the PLINK log-files for future reference. In order to keep the data directory tidy, we’ll create a directory for the log files and move them to the log directory here after each analysis step.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">refdir=</span><span class="st">'~/reference'</span>
<span class="fu">mkdir</span> -r <span class="va">$qcdir</span>/plink_log</code></pre></div>
</div>
<div id="plink-software" class="section level2">
<h2 class="hasAnchor">
<a href="#plink-software" class="anchor"></a>PLINK software</h2>
<p>In addition to <a href="https://www.cog-genomics.org/plink/1.9/">PLINK v1.9</a>, which is a requirment for the <code>plinkQC</code> package, we will also need <a href="https://www.cog-genomics.org/plink/2.0/">PLINK v2</a> for processing the downloaded the dataset. In the following, when <code>plink</code> is invoked, this corresponds to v1.9, whereas <code>plink2</code> corresponds to v2.</p>
</div>
<div id="download-and-decompress-1000-genomes-phase-3-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-and-decompress-1000-genomes-phase-3-data" class="anchor"></a>Download and decompress 1000 Genomes phase 3 data</h2>
<p>1000 Genomes phase III (1000GenomesIII) is available in <a href="https://www.cog-genomics.org/plink/2.0/input#pgen">PLINK 2 binary format</a> at <a href="https://www.cog-genomics.org/plink/2.0/resources#1kg_phase3" class="uri">https://www.cog-genomics.org/plink/2.0/resources#1kg_phase3</a>. In addition, a sample file with information about the individuals’ ancestry is available and shoud be downloaded as in input for <code>plinkQC::chec_Ancestry()</code>. The following code chunk downloads and decompresses the data. The genome build of these files is the same as the original release of the 1000GenomesIII, namely CGRCh37.</p>
<p>NB: CGRCh38 positions in vcf format can be found <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/GRCh38_positions/">here</a> The remainder of this vignette will however look at the data processing required for the 1000GenomesIII available in <a href="https://www.cog-genomics.org/plink/2.0/input#pgen">PLINK 2 binary format</a>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="bu">cd</span> <span class="va">$refdir</span>

<span class="va">pgen=</span>https://www.dropbox.com/s/afvvf1e15gqzsqo/all_phase3.pgen.zst<span class="ex">?dl</span>=1
<span class="va">pvar=</span>https://www.dropbox.com/s/0nz9ey756xfocjm/all_phase3.pvar.zst<span class="ex">?dl</span>=1
<span class="va">sample=</span>https://www.dropbox.com/s/yozrzsdrwqej63q/phase3_corrected.psam<span class="ex">?dl</span>=1

<span class="fu">wget</span> <span class="va">$pgen</span>
<span class="fu">mv</span> <span class="st">'all_phase3.pgen.zst?dl=1'</span> all_phase3.pgen.zst
<span class="ex">plink2</span> --zst-decompress all_phase3.pgen.zst <span class="op">&gt;</span> all_phase3.pgen

<span class="fu">wget</span> <span class="va">$pvar</span>
<span class="fu">mv</span> <span class="st">'all_phase3.pvar.zst?dl=1'</span> all_phase3.pvar.zst
<span class="ex">plink2</span> --zst-decompress all_phase3.pvar.zst <span class="op">&gt;</span> all_phase3.pvar

<span class="fu">wget</span> <span class="va">$sample</span>
<span class="fu">mv</span> <span class="st">'phase3_corrected.psam\?dl\=1'</span> all_phase3.psam</code></pre></div>
</div>
<div id="convert-1000-genomes-phase-3-data-to-plink-1-binary-format" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-1000-genomes-phase-3-data-to-plink-1-binary-format" class="anchor"></a>Convert 1000 Genomes phase 3 data to plink 1 binary format</h2>
<p>We then convert the <a href="https://www.cog-genomics.org/plink/2.0/input#pgen">PLINK 2 binary format</a> to the (at the moment) more standardly used <a href="https://www.cog-genomics.org/plink/1.9/inputbped">PLINK 1 binary format</a>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink2</span> --pfile <span class="va">$refdir</span>/all_phase3 \
      --make-bed \
      --out <span class="va">$refdir</span>/all_phase3
<span class="fu">mv</span> <span class="va">$refdir</span>/all_phase3.log <span class="va">$refdir</span>/log</code></pre></div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-a1000Genomes2015">
<p>1. 1000 Genomes Project Consortium. An integrated map of structural variation in 2,504 human genomes. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a></p>
</div>
<div id="ref-b1000Genomes2015">
<p>2. 1000 Genomes Project Consortium. A global reference for human genetic variation. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15393">10.1038/nature15393</a></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#set-up">Set-up</a></li>
      <li><a href="#plink-software">PLINK software</a></li>
      <li><a href="#download-and-decompress-1000-genomes-phase-3-data">Download and decompress 1000 Genomes phase 3 data</a></li>
      <li><a href="#convert-1000-genomes-phase-3-data-to-plink-1-binary-format">Convert 1000 Genomes phase 3 data to plink 1 binary format</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
