<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing HapMap III reference data for ancestry estimation • plinkQC</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing HapMap III reference data for ancestry estimation">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/1000Genomes.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Processing HapMap III reference data for ancestry estimation</h1>
                        <h4 class="author">Hannah Meyer</h4>
            
            <h4 class="date">2018-10-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HannahVMeyer/plinkQC/blob/master/vignettes/HapMap.Rmd"><code>vignettes/HapMap.Rmd</code></a></small>
      <div class="hidden name"><code>HapMap.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Genotype quality control for genetic association studies often includes the need for selecting samples of the same ethnic background. To identify individuals of divergent ancestry based on genotypes, the genotypes of the study population can be combined with genotypes of a reference dataset consisting of individuals from known ethnicities. Principal component analysis (PCA) on this combined genotype panel can then be used to detect population structure down to the level of the reference dataset.</p>
<p>The following vignette shows the processing steps required to use samples of the HapMap study <span class="citation">[1]</span><span class="citation">[2]</span><span class="citation">[3]</span> as a reference dataset. Using this reference, population structure down to large-scale continental ancestry can be detected. A step-by-step instruction on how to conduct this analysis is described in this <a href="https://hannahvmeyer.github.io/plinkQC/articles/AncestryCheck.html">vignette</a>.</p>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<div id="set-up" class="section level2">
<h2 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set-up</h2>
<p>We will first set up some bash variables and create directories needed; storing the names and directories of the reference will make it easy to use updated versions of the reference in the future. Is is also useful to keep the PLINK log-files for future reference. In order to keep the data directory tidy, we’ll create a directory for the log files and move them to the log directory here after each analysis step.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">refdir=</span><span class="st">'~/reference'</span>
<span class="fu">mkdir</span> -r <span class="va">$qcdir</span>/plink_log</code></pre></div>
</div>
<div id="download-and-convert-hapmap-phase-iii-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-and-convert-hapmap-phase-iii-data" class="anchor"></a>Download and convert Hapmap phase III data</h2>
<p>Hapmap phase 3 data (HapMapIII) is available in <a href="https://www.cog-genomics.org/plink/1.9/input#ped">PLINK text format</a> at ncbi. In addition, a sample file with information about the individuals’ ancestry is available and shoud be downloaded as in input for <code>plinkQC::chec_Ancestry()</code>. The following code chunk downloads and unzips the data.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="bu">cd</span> <span class="va">$refdir</span>

<span class="va">ftp=</span>ftp://ftp.ncbi.nlm.nih.gov/hapmap/genotypes/2009-01_phaseIII/plink_format/
<span class="va">prefix=</span>hapmap3_r2_b36_fwd.consensus.qc.poly

<span class="fu">wget</span> <span class="va">$ftp</span>/<span class="va">$prefix</span>.map.bz2
<span class="fu">bunzip2</span> <span class="va">$prefix</span>.map.bz2

<span class="fu">wget</span> <span class="va">$ftp</span>/<span class="va">$prefix</span>.ped.bz2
<span class="fu">bunzip2</span> <span class="va">$prefix</span>.per.bz2

<span class="fu">wget</span> <span class="va">$ftp</span>/relationships_w_pops_121708.txt</code></pre></div>
<p>We then convert the <a href="https://www.cog-genomics.org/plink/1.9/input#ped">PLINK text format</a> to the standardly used <a href="https://www.cog-genomics.org/plink/1.9/inputbped">PLINK binary format</a>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --file <span class="va">$refdir</span>/<span class="va">$prefix</span> \
      --make-bed \
      --out <span class="va">$refdir</span>/HapMapIII_NCBI36
<span class="fu">mv</span> <span class="va">$refdir</span>/HapMapIII_NCBI36.log <span class="va">$refdir</span>/log</code></pre></div>
</div>
<div id="update-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#update-annotation" class="anchor"></a>Update annotation</h2>
<p>The genome build of HapMap III data is NCBI36. Currently most datasets are updated to CGRCh37 or CGRCh38. In order to update the HapMap III data to the desired build, we use the UCSC <a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver">liftOver</a> tool. The liftOver tool takes information in a format similar to the <a href="https://www.cog-genomics.org/plink/1.9/formats#bim">PLINK .bim</a> format, the <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC bed format</a> and a <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/">liftover chain</a>, containing the mapping information between the old genome (target) and new genome (query). It returns the updated annotation (newFile) and a file with unmappable variants (unMapped):</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">liftOver</span> oldFile liftover.chain newFile unMapped</code></pre></div>
<p>We first need to download the liftOver tool from <a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver" class="uri">https://genome.ucsc.edu/cgi-bin/hgLiftOver</a> and the appropriate liftover chain from <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/" class="uri">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/</a>). We then convert the <a href="https://www.cog-genomics.org/plink/1.9/formats#bim">PLINK .bim</a> format, to the zero-based <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC bed</a> format.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">awk</span> <span class="st">'{print "chr" $1, $4 -1, $4, $2 }'</span> <span class="va">$refdir</span>/HapMapIII_NCBI36.bim <span class="kw">|</span> <span class="kw">\</span>
    <span class="kw">{</span> <span class="fu">sed</span> <span class="st">'s/chr23/chrX/'</span><span class="kw">;</span> <span class="fu">sed</span> <span class="st">'s/chr24/chrY/'</span><span class="kw">}</span> <span class="op">&gt;</span> <span class="kw">\</span>
    <span class="va">$refdir</span><span class="ex">/HapMapIII_NCBI36.tolift</span></code></pre></div>
<p>We use the liftOver tool and the UCSC bed formated annotation file together with the appropriate chain file to do the lift over.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">liftOver</span> <span class="va">$refdir</span>/HapMapIII_NCBI36.tolift <span class="va">$refdir</span>/hg18ToHg19.over.chain \
    <span class="va">$refdir</span>/HapMapIII_CGRCh37 <span class="va">$refdir</span>/HapMapIII_NCBI36.unMapped</code></pre></div>
<p>After successful liftover, we will be able to extract i) the variants that were mappable from the old to the new genome and ii) their updated positions</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># ectract mapped variants</span>
<span class="fu">awk</span> <span class="st">'{print $4}'</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37 <span class="op">&gt;</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37.snps
<span class="co"># ectract updated positions</span>
<span class="fu">awk</span> <span class="st">'{print $4, $3}'</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37 <span class="op">&gt;</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37.pos</code></pre></div>
</div>
<div id="update-the-reference-data" class="section level2">
<h2 class="hasAnchor">
<a href="#update-the-reference-data" class="anchor"></a>Update the reference data</h2>
<p>We can now use PLINK to extract the mappable variants from the old build and update their position. After these steps, the HapMap III dataset can be used for infering study ancestry as described in the corresponding <a href="https://hannahvmeyer.github.io/plinkQC/articles/AncestryCheck.html">vignette</a>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">plink</span> --bfile <span class="va">$refdir</span>/HapMapIII_NCBI36
    <span class="ex">--extract</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37.snps \
    --update-map <span class="va">$refdir</span>/HapMapIII_CGRCh37.pos \
    --make-bed \
    --out <span class="va">$refdir</span>/HapMapIII_CGRCh37
<span class="fu">mv</span> <span class="va">$refdir</span>/HapMapIII_CGRCh37.log <span class="va">$refdir</span>/log</code></pre></div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-HapMap2005">
<p>1. The International HapMap Consortium. A haplotype map of the human genome. Nature. 2005;437: 1299–320. doi:<a href="https://doi.org/10.1038/nature04226">10.1038/nature04226</a></p>
</div>
<div id="ref-HapMap2007">
<p>2. The International HapMap Consortium. A second generation human haplotype map of over 3.1 million SNPs. Nature. 2007;449: 851. doi:<a href="https://doi.org/10.1038/nature06258">10.1038/nature06258</a></p>
</div>
<div id="ref-HapMap2010">
<p>3. The International HapMap Consortium. Integrating common and rare genetic variation in diverse human populations. Nature. 2010;467. doi:<a href="https://doi.org/10.1038/nature09298">10.1038/nature09298</a></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#set-up">Set-up</a></li>
      <li><a href="#download-and-convert-hapmap-phase-iii-data">Download and convert Hapmap phase III data</a></li>
      <li><a href="#update-annotation">Update annotation</a></li>
      <li><a href="#update-the-reference-data">Update the reference data</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
