<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genotype quality control with plinkQC • plinkQC</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genotype quality control with plinkQC">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/1000Genomes.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Genotype quality control with plinkQC</h1>
                        <h4 class="author">Hannah Meyer</h4>
            
            <h4 class="date">2018-10-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HannahVMeyer/plinkQC/blob/master/vignettes/plinkQC.Rmd"><code>vignettes/plinkQC.Rmd</code></a></small>
      <div class="hidden name"><code>plinkQC.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Genotyping arrays enable the direct measurement of an individuals genotype at thousands of markers. Subsequent analyses such as genome-wide association studies rely on the high quality of these marker genotypes.</p>
<p>Anderson and colleagues introduced a protocol for data quality control in genetic association studies heavily based on the summary statistics and relatedness estimation functions in the PLINK software <span class="citation">[1]</span>. PLINK is a comprehensive, open-source command-line tool for genome-wide association studies (GWAS) and population genetics research <span class="citation">[2]</span>. It’s main functionalities include data management, computing individual- and marker- level summary statistics, identity-by-state estimation and association analysis.<br>
Integration with R is achieved through its R plugin or PLINK/SEQ R Package <span class="citation">[4]</span>. While the plugin is limited to operations yielding simple genetic marker vectors as output, the PLINK/SEQ R Package is limited in the functionalities it can access.</p>
<p><em>plinkQC</em> facilitates genotype quality control for genetic association studies as described by <span class="citation">[1]</span>. It wraps around plink basic statistics (e.g. missing genotyping rates per individual, allele frequencies per genetic marker) and relationship functions and generates a per-individual and per-marker quality control report. Individuals and markers that fail the quality control can subsequently be removed with <em>plinkQC</em> to generate a new, clean dataset. Removal of individuals based on relationship status is optimised to retain as many individuals as possible in the study.</p>
<p><em>plinkQC</em> depends on the <a href="https://www.cog-genomics.org/plink/1.9/">PLINK</a> (version <span class="math inline">\(\ge\)</span> 1.9), which has to be manually installed prior to the usage of <em>plinkQC</em>. It assumes the genotype have already been determined from the original probe intensity data of the genotype array and is available in <a href="https://www.cog-genomics.org/plink/1.9/input#bed">plink format</a>.</p>
<p>The protocol is implemented in three main functions, the per-individual quality control (<code>perIndividualQC</code>), the per-marker quality control (<code>perMarkerQC</code>) and the generation of the new, quality control dataset (<code>cleanData</code>):</p>
<div id="per-individual-quality-control" class="section level2">
<h2 class="hasAnchor">
<a href="#per-individual-quality-control" class="anchor"></a>Per-individual quality control</h2>
<p>The per-individual quality control with <code>perIndividualQC</code> wraps around these functions: (i) <code>check_sex</code>: for the identification of individuals with discordant sex information, (ii) <code>check_heterozygosity_and_missingness</code>: for the identification of individuals with outlying missing genotype and/or heterozygosity rates, (iii) <code>check_relatedness</code>: for the identification of related individuals, (iv) <code>check_ancestry</code>: identification of individuals of divergent ancestry.</p>
</div>
<div id="per-marker-quality-control" class="section level2">
<h2 class="hasAnchor">
<a href="#per-marker-quality-control" class="anchor"></a>Per-marker quality control</h2>
<p>The per-marker quality control with <code>perMarkerQC</code> wraps around these functions: (i) <code>check_snp_missingnes</code>: for the identifying markers with excessive missing genotype rates, (ii) <code>check_hwe</code>: for the identifying markers showing a significant deviation from Hardy-Weinberg equilibrium (HWE), (iii) <code>check_maf</code>: for the removal of markers with low minor allele frequency (MAF).</p>
</div>
<div id="clean-data" class="section level2">
<h2 class="hasAnchor">
<a href="#clean-data" class="anchor"></a>Clean data</h2>
<p><code>cleanData</code> takes the results of <code>perMarkerQC</code> and <code>perIndividualQC</code> and creates a new dataset with all individuals and markers that passed the quality control checks.</p>
</div>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<p>In the following, genotype quality control with <em>plinkQC</em> is applied on a small example dataset with 200 individuals and 10,000 markers (provided with this package). The quality control is demonstrated in three easy steps, per-individual and per-marker quality control followed by the generation of the new dataset. In addition, the functionality of each of the functions underlying <code>perMarkerQC</code> and <code>perIndividualQC</code> is demonstrated at the end of this vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">package.dir &lt;-<span class="st"> </span><span class="kw">find.package</span>(<span class="st">'plinkQC'</span>)
qcdir &lt;-<span class="st"> </span><span class="kw">file.path</span>(package.dir, <span class="st">'extdata'</span>)
alg &lt;-<span class="st"> 'data'</span>
path2plink &lt;-<span class="st"> "~/bin"</span></code></pre></div>
<div id="per-individual-quality-control-1" class="section level2">
<h2 class="hasAnchor">
<a href="#per-individual-quality-control-1" class="anchor"></a>Per-individual quality control</h2>
<p>For <code>perIndividualQC</code>, one simply specifies the directory where the data is stored (qcdir) and the prefix of the plink files (i.e. prefix.bim, prefix.bed, prefix.fam). In addition, the names of the files containing information about the reference population and the merged dataset used in <code>check_ancestry</code> have to be provided: refSamplesFile, refColorsFile and prefixMergedDataset. Per default, all quality control checks will be conducted.</p>
<p>In addition to running each check, <code>perIndividualQC</code> writes a list of all fail individual IDs to the qcdir. These IDs will be removed in the computation of the <code>perMarkerQC</code>. If the list is not present, <code>perMarkerQC</code> will send a message about conducting the quality control on the entire dataset.</p>
<p><code>perIndividualQC</code> displays the results of the quality control steps in a multi-panel plot. NB: To reduce the data size of the example data in plinkQC, data.genome has already been reduced to the individuals that are related. Thus the relatedness plots in C only show counts for related individuals.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_individuals &lt;-<span class="st"> </span><span class="kw"><a href="../reference/perIndividualQC.html">perIndividualQC</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg,
                            <span class="dt">refSamplesFile=</span><span class="kw">paste</span>(qcdir, <span class="st">"/HapMap_ID2Pop.txt"</span>,
                                                 <span class="dt">sep=</span><span class="st">""</span>), 
                            <span class="dt">refColorsFile=</span><span class="kw">paste</span>(qcdir, <span class="st">"/HapMap_PopColors.txt"</span>,
                                                 <span class="dt">sep=</span><span class="st">""</span>),
                            <span class="dt">prefixMergedDataset=</span><span class="st">"data.HapMapIII"</span>,
                            <span class="dt">interactive=</span><span class="ot">TRUE</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>,
                            <span class="dt">do.run_check_het_and_miss=</span><span class="ot">FALSE</span>,
                            <span class="dt">do.run_check_sex=</span><span class="ot">FALSE</span>,
                            <span class="dt">do.run_check_relatedness=</span><span class="ot">FALSE</span>,
                            <span class="dt">do.run_check_ancestry=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Identification of individuals with discordant sex information</span>
<span class="co">#&gt; Identification of individuals with outlying missing genotype or heterozygosity rates</span>
<span class="co">#&gt; Identification of related individuals</span>
<span class="co">#&gt; Identification of individuals of divergent ancestry</span>
<span class="co">#&gt; Combine fail IDs into /Library/Frameworks/R.framework/Versions/3.5/Resources/library/plinkQC/extdata/data.fail.IDs</span></code></pre></div>
<p><img src="plinkQC_files/figure-html/individual%20QC-1.png" width="864"></p>
<p><code>overviewperIndividualQC</code> depicts overview plots of quality control failures and the intersection of quality control failures with ancestry exclusion.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">overview_individuals &lt;-<span class="st"> </span><span class="kw"><a href="../reference/overviewPerIndividualQC.html">overviewPerIndividualQC</a></span>(fail_individuals,
                                                <span class="dt">interactive=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="plinkQC_files/figure-html/overview%20individual%20QC-1.png" width="700"><img src="plinkQC_files/figure-html/overview%20individual%20QC-2.png" width="700"></p>
</div>
<div id="per-marker-quality-control-1" class="section level2">
<h2 class="hasAnchor">
<a href="#per-marker-quality-control-1" class="anchor"></a>Per-marker quality control</h2>
<p><code>perMarkerQC</code> applies its checks to data in the specified directory (qcdir), starting with the specified prefix of the plink files (i.e. prefix.bim, prefix.bed, prefix.fam). Optionally, the user can specify different thresholds for the quality control checks and which check to conduct. Per default, all quality control checks will be conducted. <code>perMarkerQC</code> displays the results of the QC step in a multi-panel plot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_markers &lt;-<span class="st"> </span><span class="kw"><a href="../reference/perMarkerQC.html">perMarkerQC</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">path2plink=</span>path2plink,
                            <span class="dt">verbose=</span><span class="ot">TRUE</span>, <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                            <span class="dt">showPlinkOutput=</span><span class="ot">FALSE</span>)</code></pre></div>
<embed src="markerQC.pdf" style="display: block; margin: auto;" type="application/pdf"></embed><p><code>overviewPerMarkerQC</code> depicts an overview of the marker quality control failures and their overlaps.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">overview_marker &lt;-<span class="st"> </span><span class="kw"><a href="../reference/overviewPerMarkerQC.html">overviewPerMarkerQC</a></span>(fail_markers, <span class="dt">interactive=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><embed src="overviewMarkerQC.pdf"></embed><!-- --></p>

</div>
<div id="create-qc-ed-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#create-qc-ed-dataset" class="anchor"></a>Create QC-ed dataset</h2>
<p>After checking results of the per-individual and per-marker quality control, individuals and markers that fail the chosen criteria can automatically be removed from the dataset with <code>cleanData</code>, resulting in the new dataset qcdir/data.clean.bed,qcdir/data.clean.bim, qcdir/data.clean.fam. For convenience, <code>cleanData</code> returns a list of all individuals in the study split into keep and remove individuals.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Ids  &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cleanData.html">cleanData</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">path2plink=</span>path2plink,
                            <span class="dt">verbose=</span><span class="ot">TRUE</span>, <span class="dt">showPlinkOutput=</span><span class="ot">FALSE</span>)</code></pre></div>
</div>
</div>
<div id="step-by-step" class="section level1">
<h1 class="hasAnchor">
<a href="#step-by-step" class="anchor"></a>Step-by-step</h1>
<div id="individuals-with-discordant-sex-information" class="section level2">
<h2 class="hasAnchor">
<a href="#individuals-with-discordant-sex-information" class="anchor"></a>Individuals with discordant sex information</h2>
<p>The identification of individuals with discordant sex information helps to detect sample mix-ups and samples with very poor genotyping rates. For each sample, the homozygosity rates across all X-chromosomal genetic markers are computed and compared with the expected rates (typically $&lt;$0.2 for females and $&gt;$0.8 for males). For samples where the assigned sex (PEDSEX in the .fam file) contradicts the sex inferred from the homozygosity rates (SNPSEX), it should be checked that the sex was correctly recorded (genotyping often occurs at different locations as phenotyping and misrecording might occur). Samples with discordant sex information that is not accounted for should be removed from the study. Identifying individuals with discordant sex information is implemented in <code>check_sex</code>. It finds individuals whose SNPSEX != PEDSEX. Optionally, an extra data.frame with sample IDs and sex can be provided to double check if external and PEDSEX data (often processed at different centers) match. If a mismatch between PEDSEX and SNPSEX was detected, by SNPSEX == Sex, PEDSEX of these individuals can optionally be updated. <code>check_sex</code> depicts the X-chromosomal heterozygosity (SNPSEX) of the samples split by their (PEDSEX).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_sex &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_sex.html">check_sex</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">interactive=</span><span class="ot">TRUE</span>, <span class="dt">verbose=</span><span class="ot">TRUE</span>,
                      <span class="dt">run.check_sex=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="plinkQC_files/figure-html/check%20sex-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="individuals-with-outlying-missing-genotype-andor-heterozygosity-rates" class="section level2">
<h2 class="hasAnchor">
<a href="#individuals-with-outlying-missing-genotype-andor-heterozygosity-rates" class="anchor"></a>Individuals with outlying missing genotype and/or heterozygosity rates</h2>
<p>The identification of individuals with outlying missing genotype and/or heterozygosity rates helps to detect samples with poor DNA quality and/or concentration that should be excluded from the study. Typically, individuals with more than 3-7% of their genotype calles missing are removed. Outlying heterozygosity rates are judged relative to the overall heterozygosity rates in the study, and individuals whose rates are more than a few standard deviations (sd) from the mean heterozygosity rate are removed. A typical quality control for outlying heterozygoisty rates would remove individuals who are three sd away from the mean rate. Identifying related individuals with outlying missing genotype and/or heterozygosity rates is implemented in <code>check_het_and_miss</code>. It finds individuals that have genotyping and heterozygosity rates that fail the set thresholds and depicts the results as a scatter plot with the samples’ missingness rates on x-axis and their heterozygosity rates on the y-axis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_het_imiss &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_het_and_miss.html">check_het_and_miss</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg,
                                     <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                                     <span class="dt">run.check_het_and_miss=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="plinkQC_files/figure-html/check%20het%20miss-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="related-individuals" class="section level2">
<h2 class="hasAnchor">
<a href="#related-individuals" class="anchor"></a>Related individuals</h2>
<p>Depending on the future use of the genotypes, it might required to remove any related individuals from the study. Related individuals can be identified by their proporting of shared alleles at the genotyped markers (identity by descend, IBD). Standardly, individuals with second-degree relatedness or higher will be excluded. Identifying related individuals is implemented in <code>check_relatedness</code>. It finds pairs of samples whose proportion of IBD is larger than the specified highIBDTh. Subsequently, for pairs of individual that do not have additional relatives in the dataset, the individual with the greater genotype missingness rate is selected and returned as the individual failing the relatedness check. For more complex family structures, the unrelated individuals per family are selected (e.g. in a parents-offspring trio, the offspring will be marked as fail, while the parents will be kept in the analysis).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exclude_relatedness &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_relatedness.html">check_relatedness</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                                         <span class="dt">run.check_relatedness=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="plinkQC_files/figure-html/check%20related-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="individuals-of-divergent-ancestry" class="section level2">
<h2 class="hasAnchor">
<a href="#individuals-of-divergent-ancestry" class="anchor"></a>Individuals of divergent ancestry</h2>
<p>The identification of individuals of divergent ancestry can be achieved by combining the genotypes of the study population with genotypes of a reference dataset consisting of individuals from known ethnicities (for instance individuals from the Hapmap or 1000 genomes study <span class="citation">[5]</span>). Principal component analysis on this combined genotype panel can be used to detect population structure down to the level of the reference dataset (for Hapmap and 1000 Genomes, this is down to large-scale continental ancestry). Identifying individuals of divergent ancestry is implemented in <code>check_ancestry</code>. Currently, check ancestry only supports automatic selection of individuals of European descent. It uses information from principal components 1 and 2 to find the center of the European reference samples. All study samples whose euclidean distance from the centre falls outside a specified radius are considered non-European. <code>check_ancestry</code> creates a scatter plot of PC1 versus PC2 color-coded for samples of the reference populations and the study population.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">exclude_ancestry &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_ancestry.html">check_ancestry</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg,
                            <span class="dt">refSamplesFile=</span><span class="kw">paste</span>(qcdir, <span class="st">"/HapMap_ID2Pop.txt"</span>,
                                                 <span class="dt">sep=</span><span class="st">""</span>), 
                            <span class="dt">refColorsFile=</span><span class="kw">paste</span>(qcdir, <span class="st">"/HapMap_PopColors.txt"</span>,
                                                 <span class="dt">sep=</span><span class="st">""</span>),
                            <span class="dt">prefixMergedDataset=</span><span class="st">"data.HapMapIII"</span>,
                            <span class="dt">run.check_ancestry=</span><span class="ot">FALSE</span>,
                            <span class="dt">interactive=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="plinkQC_files/figure-html/check%20ancestry-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="markers-with-excessive-missingness-rate" class="section level2">
<h2 class="hasAnchor">
<a href="#markers-with-excessive-missingness-rate" class="anchor"></a>Markers with excessive missingness rate</h2>
<p>Markers with excessive missingness rate are removed as they are considered unreliable. Typically, thresholds for marker exclusion based on missingness range from 1%-5%. Identifying markers with high missingness rates is implemented in <code>snp_missingness</code>. It calculates the rates of missing genotype calls and frequency for all variants in the individuals that passed the <code>perIndividualQC</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_snpmissing &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_snp_missingness.html">check_snp_missingness</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                                         <span class="dt">path2plink=</span>path2plink, 
                                         <span class="dt">showPlinkOutput=</span><span class="ot">FALSE</span>)</code></pre></div>
<embed src="snpmissingness.pdf" style="display: block; margin: auto;" type="application/pdf"></embed>
</div>
<div id="markers-with-deviation-from-hwe" class="section level2">
<h2 class="hasAnchor">
<a href="#markers-with-deviation-from-hwe" class="anchor"></a>Markers with deviation from HWE</h2>
<p>Markers with strong deviation from HWE might be indicative of genotyping or genotype-calling errors. As serious genotyping errors often yield very low p-values (in the order of <span class="math inline">\(10^-50\)</span>), it is recommended to choose a reasonably low threshold to avoid filtering too many variants (that might have slight, non-critical deviations). Identifying markers with deviation from HWE is implemented in <code>check_hwe</code>. It calculates the observed and expected heterozygote frequencies per SNP in the individuals that passed the <code>perIndividualQC</code> and computes the deviation of the frequencies from Hardy-Weinberg equilibrium (HWE) by HWE exact test.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_hwe &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_hwe.html">check_hwe</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                      <span class="dt">path2plink=</span>path2plink, <span class="dt">showPlinkOutput=</span><span class="ot">FALSE</span>)</code></pre></div>
<embed src="hwe.pdf" style="display: block; margin: auto;" type="application/pdf"></embed>
</div>
<div id="markers-with-low-minor-allele-frequency" class="section level2">
<h2 class="hasAnchor">
<a href="#markers-with-low-minor-allele-frequency" class="anchor"></a>Markers with low minor allele frequency</h2>
<p>Markers with low minor allele count are often removed as the actual genotype calling (via the calling algorithm) is very difficult due to the small sizes of the heterozygote and rare-homozygote clusters. Identifying markers with low minor allele count is implemented in <code>check_maf</code>. It calculates the minor allele frequencies for all variants in the individuals that passed the <code>perIndividualQC</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fail_maf &lt;-<span class="st"> </span><span class="kw"><a href="../reference/check_maf.html">check_maf</a></span>(<span class="dt">qcdir=</span>qcdir, <span class="dt">alg=</span>alg, <span class="dt">interactive=</span><span class="ot">TRUE</span>,
                      <span class="dt">path2plink=</span>path2plink, <span class="dt">showPlinkOutput=</span><span class="ot">FALSE</span>)</code></pre></div>
<embed src="maf.pdf" style="display: block; margin: auto;" type="application/pdf"></embed>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Anderson2010">
<p>1. Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT. Data quality control in genetic case-control association studies. Nature Protocols. 2010;5: 1564–73. doi:<a href="https://doi.org/10.1038/nprot.2010.116">10.1038/nprot.2010.116</a></p>
</div>
<div id="ref-Purcell2007">
<p>2. Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, et al. PLINK: a tool set for whole-genome association and population-based linkage analyses. The American Journal of Human Genetics. 2007;81: 559–75. doi:<a href="https://doi.org/10.1086/519795">10.1086/519795</a></p>
</div>
<div id="ref-Chang2015">
<p>3. Chang CC, Chow CC, Tellier LC, Vattikuti S, Purcell SM, Lee JJ. Second-generation PLINK: rising to the challenge of larger and richer datasets. GigaScience. 2015;4: 7. doi:<a href="https://doi.org/10.1186/s13742-015-0047-8">10.1186/s13742-015-0047-8</a></p>
</div>
<div id="ref-PlinkSeq2014">
<p>4. PLINK/SEQ. [Internet]. 2014. Available: <a href="https://atgu.mgh.harvard.edu/plinkseq/index.shtml" class="uri">https://atgu.mgh.harvard.edu/plinkseq/index.shtml</a></p>
</div>
<div id="ref-HapMap2005">
<p>5. The International HapMap Consortium. A haplotype map of the human genome. Nature. 2005;437: 1299–320. doi:<a href="https://doi.org/10.1038/nature04226">10.1038/nature04226</a></p>
</div>
<div id="ref-HapMap2007">
<p>6. The International HapMap Consortium. A second generation human haplotype map of over 3.1 million SNPs. Nature. 2007;449: 851. doi:<a href="https://doi.org/10.1038/nature06258">10.1038/nature06258</a></p>
</div>
<div id="ref-HapMap2010">
<p>7. The International HapMap Consortium. Integrating common and rare genetic variation in diverse human populations. Nature. 2010;467. doi:<a href="https://doi.org/10.1038/nature09298">10.1038/nature09298</a></p>
</div>
<div id="ref-a1000Genomes2015">
<p>8. 1000 Genomes Project Consortium. An integrated map of structural variation in 2,504 human genomes. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a></p>
</div>
<div id="ref-b1000Genomes2015">
<p>9. 1000 Genomes Project Consortium. A global reference for human genetic variation. Nature. 2015;526: 75–81. doi:<a href="https://doi.org/10.1038/nature15393">10.1038/nature15393</a></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#per-individual-quality-control">Per-individual quality control</a></li>
      <li><a href="#per-marker-quality-control">Per-marker quality control</a></li>
      <li><a href="#clean-data">Clean data</a></li>
      </ul>
</li>
      <li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#per-individual-quality-control-1">Per-individual quality control</a></li>
      <li><a href="#per-marker-quality-control-1">Per-marker quality control</a></li>
      <li><a href="#create-qc-ed-dataset">Create QC-ed dataset</a></li>
      </ul>
</li>
      <li>
<a href="#step-by-step">Step-by-step</a><ul class="nav nav-pills nav-stacked">
<li><a href="#individuals-with-discordant-sex-information">Individuals with discordant sex information</a></li>
      <li><a href="#individuals-with-outlying-missing-genotype-andor-heterozygosity-rates">Individuals with outlying missing genotype and/or heterozygosity rates</a></li>
      <li><a href="#related-individuals">Related individuals</a></li>
      <li><a href="#individuals-of-divergent-ancestry">Individuals of divergent ancestry</a></li>
      <li><a href="#markers-with-excessive-missingness-rate">Markers with excessive missingness rate</a></li>
      <li><a href="#markers-with-deviation-from-hwe">Markers with deviation from HWE</a></li>
      <li><a href="#markers-with-low-minor-allele-frequency">Markers with low minor allele frequency</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
