<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Identification of individuals with discordant sex information — check_sex • plinkQC</title>

<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Identification of individuals with discordant sex information — check_sex" />

<meta property="og:description" content="Runs and evaluates results from plink --check-sex.
check_sex returns IIDs for individuals whose SNPSEX != PEDSEX
(where the SNPSEX is determined by the heterozygosity rate across
X-chromosomal variants).
Mismatching SNPSEX and PEDSEX IDs can indicate plating errors, sample-mixup
or generally samples with poor genotyping. In the latter case, these IDs are
likely to fail other QC steps as well.
Optionally, an extra data.frame (externalSex) with sample IDs and sex can be
provided to double check if external and PEDSEX data (often processed at
different centers) match. If a mismatch between PEDSEX and SNPSEX was
detected, while SNPSEX == Sex, PEDSEX of these individuals can optionally be
updated (fixMixup=TRUE).
check_sex depicts the X-chromosomal heterozygosity (SNPSEX) of
the individuals split by their (PEDSEX)." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/Genomes1000.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Identification of individuals with discordant sex information</h1>
    <small class="dont-index">Source: <a href='https://github.com/HannahVMeyer/plinkQC/blob/master/R/individualQC.R'><code>R/individualQC.R</code></a></small>
    <div class="hidden name"><code>check_sex.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Runs and evaluates results from plink --check-sex.
<code>check_sex</code> returns IIDs for individuals whose SNPSEX != PEDSEX
(where the SNPSEX is determined by the heterozygosity rate across
X-chromosomal variants).
Mismatching SNPSEX and PEDSEX IDs can indicate plating errors, sample-mixup
or generally samples with poor genotyping. In the latter case, these IDs are
likely to fail other QC steps as well.
Optionally, an extra data.frame (externalSex) with sample IDs and sex can be
provided to double check if external and PEDSEX data (often processed at
different centers) match. If a mismatch between PEDSEX and SNPSEX was
detected, while SNPSEX == Sex, PEDSEX of these individuals can optionally be
updated (fixMixup=TRUE).
<code>check_sex</code> depicts the X-chromosomal heterozygosity (SNPSEX) of
the individuals split by their (PEDSEX).</p>
    
    </div>

    <pre class="usage"><span class='fu'>check_sex</span>(<span class='no'>indir</span>, <span class='no'>name</span>, <span class='kw'>qcdir</span> <span class='kw'>=</span> <span class='no'>indir</span>, <span class='kw'>maleTh</span> <span class='kw'>=</span> <span class='fl'>0.8</span>, <span class='kw'>femaleTh</span> <span class='kw'>=</span> <span class='fl'>0.2</span>,
  <span class='kw'>run.check_sex</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>externalSex</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>externalFemale</span> <span class='kw'>=</span> <span class='st'>"F"</span>,
  <span class='kw'>externalMale</span> <span class='kw'>=</span> <span class='st'>"M"</span>, <span class='kw'>externalSexSex</span> <span class='kw'>=</span> <span class='st'>"Sex"</span>, <span class='kw'>externalSexID</span> <span class='kw'>=</span> <span class='st'>"IID"</span>,
  <span class='kw'>fixMixup</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>interactive</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>label</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>path2plink</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>showPlinkOutput</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>indir</th>
      <td><p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam and name.sexcheck.</p></td>
    </tr>
    <tr>
      <th>qcdir</th>
      <td><p>[character] /path/to/directory to save name.sexcheck as returned
by plink --check-sex. Per default qcdir=indir. If run.check_sex is FALSE, it
is assumed that plink --check-sex has been run and qcdir/name.sexcheck is
present. User needs writing permission to qcdir.</p></td>
    </tr>
    <tr>
      <th>maleTh</th>
      <td><p>[double] Threshold of X-chromosomal heterozygosity rate for
males.</p></td>
    </tr>
    <tr>
      <th>femaleTh</th>
      <td><p>[double] Threshold of X-chromosomal heterozygosity rate for
females.</p></td>
    </tr>
    <tr>
      <th>run.check_sex</th>
      <td><p>[logical] Should plink --check-sex be run? if set to
FALSE, it is assumed that plink --check-sex has been run and
qcdir/name.sexcheck is present; <code>check_sex</code>  will fail with
missing file error otherwise.</p></td>
    </tr>
    <tr>
      <th>externalSex</th>
      <td><p>[data.frame, optional] Dataframe with sample IDs
[externalSexID] and sex [externalSexSex] to double check if external and
PEDSEX data (often processed at different centers) match.</p></td>
    </tr>
    <tr>
      <th>externalFemale</th>
      <td><p>[integer/character] Identifier for 'female' in
externalSex.</p></td>
    </tr>
    <tr>
      <th>externalMale</th>
      <td><p>[integer/character] Identifier for 'male' in externalSex.</p></td>
    </tr>
    <tr>
      <th>externalSexSex</th>
      <td><p>[character] Column identifier for column containing sex
information in externalSex.</p></td>
    </tr>
    <tr>
      <th>externalSexID</th>
      <td><p>[character] Column identifier for column containing ID
information in externalSex.</p></td>
    </tr>
    <tr>
      <th>fixMixup</th>
      <td><p>[logical] Should PEDSEX of individuals with mismatch between
PEDSEX and Sex while Sex==SNPSEX automatically corrected: this will directly
change the name.bim/.bed/.fam files!</p></td>
    </tr>
    <tr>
      <th>interactive</th>
      <td><p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_sexcheck) via ggplot2::ggsave(p=p_sexcheck,
other_arguments) or pdf(outfile) print(p_sexcheck) dev.off().</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>[logical] If TRUE, progress info is printed to standard out.</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p>[logical] Set TRUE, to add fail IDs as text labels in scatter
plot.</p></td>
    </tr>
    <tr>
      <th>path2plink</th>
      <td><p>[character] Absolute path to PLINK executable
(<a href='https://www.cog-genomics.org/plink/1.9/'>https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accesible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href='http://www.rdocumentation.org/packages/sys/topics/exec'>exec_wait</a></code>('plink').</p></td>
    </tr>
    <tr>
      <th>showPlinkOutput</th>
      <td><p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Named list with i) fail_sex: [data.frame] with FID, IID, PEDSEX,
SNPSEX and Sex (if externalSex was provided) of individuals failing sex
check, ii) mixup: dataframe with FID, IID, PEDSEX, SNPSEX and Sex (if
externalSex was provided) of individuals whose PEDSEX != Sex and Sex ==
SNPSEX and iii) p_sexcheck, a ggplot2-object 'containing' a scatter plot of
the X-chromosomal heterozygosity (SNPSEX) of the sample split by their
(PEDSEX), which can be shown by print(p_sexcheck).</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>check_sex</code> wraps around <code><a href='run_check_sex.html'>run_check_sex</a></code>  and
<code><a href='evaluate_check_sex.html'>evaluate_check_sex</a></code>. If run.check_sex is TRUE,
<code><a href='run_check_sex.html'>run_check_sex</a></code> is executed ; otherwise it is assumed that plink
--check-sex has been run externally and qcdir/name.sexcheck exists.
<code>check_sex</code>  will fail with missing file error otherwise.</p>
<p>For details on the output data.frame fail_sex, check the original
description on the PLINK output format page:
<a href='https://www.cog-genomics.org/plink/1.9/formats#sexcheck'>https://www.cog-genomics.org/plink/1.9/formats#sexcheck</a>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'> </div><span class='co'># NOT RUN {</span>
<span class='no'>indir</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"plinkQC"</span>)
<span class='no'>name</span> <span class='kw'>&lt;-</span> <span class='st'>"data"</span>
<span class='no'>fail_sex</span> <span class='kw'>&lt;-</span> <span class='fu'>check_sex</span>(<span class='kw'>indir</span><span class='kw'>=</span><span class='no'>indir</span>, <span class='kw'>name</span><span class='kw'>=</span><span class='no'>name</span>, <span class='kw'>run.check_sex</span><span class='kw'>=</span><span class='fl'>FALSE</span>,
<span class='kw'>interactive</span><span class='kw'>=</span><span class='fl'>FALSE</span>, <span class='kw'>verbose</span><span class='kw'>=</span><span class='fl'>FALSE</span>)
<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

