<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create plink dataset with individuals and markers passing quality control — cleanData • plinkQC</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Create plink dataset with individuals and markers passing quality control — cleanData" />
<meta property="og:description" content="Individuals that fail per-individual QC and markers that fail
per-marker QC are removed from indir/name.bim/.bed/.fam and a new, dataset
with the remaining individuals and markers is created as
qcdir/name.clean.bim/.bed/.fam." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/Genomes1000.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../news/index.html">
    <span class="fas fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fas fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fas fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create plink dataset with individuals and markers passing quality control</h1>
    <small class="dont-index">Source: <a href='https://github.com/meyer-lab-cshl/plinkQC/blob/master/R/applyQC.R'><code>R/applyQC.R</code></a></small>
    <div class="hidden name"><code>cleanData.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Individuals that fail per-individual QC and markers that fail
per-marker QC are removed from indir/name.bim/.bed/.fam and a new, dataset
with the remaining individuals and markers is created as
qcdir/name.clean.bim/.bed/.fam.</p>
    </div>

    <pre class="usage"><span class='fu'>cleanData</span><span class='op'>(</span>
  <span class='va'>indir</span>,
  <span class='va'>name</span>,
  qcdir <span class='op'>=</span> <span class='va'>indir</span>,
  filterSex <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filterHeterozygosity <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filterSampleMissingness <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filterAncestry <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filterRelated <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filterSNPMissingness <span class='op'>=</span> <span class='cn'>TRUE</span>,
  lmissTh <span class='op'>=</span> <span class='fl'>0.01</span>,
  filterHWE <span class='op'>=</span> <span class='cn'>TRUE</span>,
  hweTh <span class='op'>=</span> <span class='fl'>1e-05</span>,
  filterMAF <span class='op'>=</span> <span class='cn'>TRUE</span>,
  macTh <span class='op'>=</span> <span class='fl'>20</span>,
  mafTh <span class='op'>=</span> <span class='cn'>NULL</span>,
  path2plink <span class='op'>=</span> <span class='cn'>NULL</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  keep_individuals <span class='op'>=</span> <span class='cn'>NULL</span>,
  remove_individuals <span class='op'>=</span> <span class='cn'>NULL</span>,
  exclude_markers <span class='op'>=</span> <span class='cn'>NULL</span>,
  extract_markers <span class='op'>=</span> <span class='cn'>NULL</span>,
  showPlinkOutput <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>indir</th>
      <td><p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p></td>
    </tr>
    <tr>
      <th>qcdir</th>
      <td><p>[character] /path/to/directory where results will be written to.
If <code><a href='perIndividualQC.html'>perIndividualQC</a></code> was conducted, this directory should be the
same as qcdir specified in <code><a href='perIndividualQC.html'>perIndividualQC</a></code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p></td>
    </tr>
    <tr>
      <th>filterSex</th>
      <td><p>[logical] Set to exclude samples that failed the sex
check (via <code><a href='check_sex.html'>check_sex</a></code> or <code><a href='perIndividualQC.html'>perIndividualQC</a></code>).
Requires file qcdir/name.fail-sexcheck.IDs (automatically created by
<code><a href='perIndividualQC.html'>perIndividualQC</a></code> if do.evaluate_check_sex set to TRUE).</p></td>
    </tr>
    <tr>
      <th>filterHeterozygosity</th>
      <td><p>[logical] Set to exclude samples that failed
check for outlying heterozygosity rates (via
<code><a href='check_het_and_miss.html'>check_het_and_miss</a></code> or
<code><a href='perIndividualQC.html'>perIndividualQC</a></code>). Requires file qcdir/name.fail-het.IDs
(automatically created by <code><a href='perIndividualQC.html'>perIndividualQC</a></code> if
do.evaluate_check_het_and_miss set to TRUE).</p></td>
    </tr>
    <tr>
      <th>filterSampleMissingness</th>
      <td><p>[logical] Set to exclude samples that failed
check for excessive missing genotype rates (via
<code><a href='check_het_and_miss.html'>check_het_and_miss</a></code> or
<code><a href='perIndividualQC.html'>perIndividualQC</a></code>). Requires file qcdir/name.fail-imiss.IDs
(automatically created by <code><a href='perIndividualQC.html'>perIndividualQC</a></code> if
do.evaluate_check_het_and_miss set to TRUE).</p></td>
    </tr>
    <tr>
      <th>filterAncestry</th>
      <td><p>[logical] Set to exclude samples that failed ancestry
check (via <code><a href='check_ancestry.html'>check_ancestry</a></code> or <code><a href='perIndividualQC.html'>perIndividualQC</a></code>).
Requires file qcdir/name.fail-ancestry.IDs (automatically created by
<code><a href='perIndividualQC.html'>perIndividualQC</a></code> if do.check_ancestry set to TRUE).</p></td>
    </tr>
    <tr>
      <th>filterRelated</th>
      <td><p>[logical] Set to exclude samples that failed relatedness
check (via <code><a href='check_relatedness.html'>check_relatedness</a></code> or <code><a href='perIndividualQC.html'>perIndividualQC</a></code>).
Requires file qcdir/name.fail-IBD.IDs (automatically created by
<code><a href='perIndividualQC.html'>perIndividualQC</a></code> if do.evaluate_check_relatedness set to TRUE).</p></td>
    </tr>
    <tr>
      <th>filterSNPMissingness</th>
      <td><p>[logical] Set to exclude markers that have
excessive missing rates across samples (via
<code><a href='check_snp_missingness.html'>check_snp_missingness</a></code> or <code><a href='perMarkerQC.html'>perMarkerQC</a></code>). Requires
lmissTh to be set.</p></td>
    </tr>
    <tr>
      <th>lmissTh</th>
      <td><p>[double] Threshold for acceptable variant missing rate across
samples.</p></td>
    </tr>
    <tr>
      <th>filterHWE</th>
      <td><p>[logical] Set to exclude markers that fail HWE exact test
(via <code><a href='check_hwe.html'>check_hwe</a></code> or <code><a href='perMarkerQC.html'>perMarkerQC</a></code>). Requires hweTh to
be set.</p></td>
    </tr>
    <tr>
      <th>hweTh</th>
      <td><p>[double] Significance threshold for deviation from HWE.</p></td>
    </tr>
    <tr>
      <th>filterMAF</th>
      <td><p>[logical] Set to exclude markers that fail minor allele
frequency or minor allele count threshold (via <code><a href='check_maf.html'>check_maf</a></code> or
<code><a href='perMarkerQC.html'>perMarkerQC</a></code>). Requires mafTh or macTh to be set.</p></td>
    </tr>
    <tr>
      <th>macTh</th>
      <td><p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p></td>
    </tr>
    <tr>
      <th>mafTh</th>
      <td><p>[double] Threshold for minor allele frequency cut-off.</p></td>
    </tr>
    <tr>
      <th>path2plink</th>
      <td><p>[character] Absolute path to PLINK executable
(<a href='https://www.cog-genomics.org/plink/1.9/'>https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href='https://rdrr.io/pkg/sys/man/exec.html'>exec</a></code>('plink').</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>[logical] If TRUE, progress info is printed to standard out.</p></td>
    </tr>
    <tr>
      <th>keep_individuals</th>
      <td><p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href='https://www.cog-genomics.org/plink/1.9/filter#indiv'>https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p></td>
    </tr>
    <tr>
      <th>remove_individuals</th>
      <td><p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href='https://www.cog-genomics.org/plink/1.9/filter#indiv'>https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p></td>
    </tr>
    <tr>
      <th>exclude_markers</th>
      <td><p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href='https://www.cog-genomics.org/plink/1.9/filter#snp'>https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p></td>
    </tr>
    <tr>
      <th>extract_markers</th>
      <td><p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href='https://www.cog-genomics.org/plink/1.9/filter#snp'>https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p></td>
    </tr>
    <tr>
      <th>showPlinkOutput</th>
      <td><p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>names [list] with i) passIDs, containing a [data.frame] with family
[FID] and individual [IID] IDs of samples that pass the QC, ii) failIDs,
containing a [data.frame] with family [FID] and individual [IID] IDs of
samples that fail the QC.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>package.dir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/find.package.html'>find.package</a></span><span class='op'>(</span><span class='st'>'plinkQC'</span><span class='op'>)</span>
<span class='va'>indir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>package.dir</span>, <span class='st'>'extdata'</span><span class='op'>)</span>
<span class='va'>qcdir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='va'>name</span> <span class='op'>&lt;-</span> <span class='st'>"data"</span>
<span class='va'>path2plink</span> <span class='op'>&lt;-</span> <span class='st'>'/path/to/plink'</span>
<span class='co'># the following code is not run on package build, as the path2plink on the</span>
<span class='co'># user system is not known.</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='co'># Run qc on all samples and markers in the dataset</span>
<span class='co'>## Run individual QC checks</span>
<span class='va'>fail_individuals</span> <span class='op'>&lt;-</span> <span class='fu'><a href='perIndividualQC.html'>perIndividualQC</a></span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>,
refSamplesFile<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>qcdir</span>, <span class='st'>"/HapMap_ID2Pop.txt"</span>,sep<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span>,
refColorsFile<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>qcdir</span>, <span class='st'>"/HapMap_PopColors.txt"</span>, sep<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span>,
prefixMergedDataset<span class='op'>=</span><span class='st'>"data.HapMapIII"</span>, interactive<span class='op'>=</span><span class='cn'>FALSE</span>, verbose<span class='op'>=</span><span class='cn'>FALSE</span>,
path2plink<span class='op'>=</span><span class='va'>path2plink</span><span class='op'>)</span>

<span class='co'>## Run marker QC checks</span>
<span class='va'>fail_markers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='perMarkerQC.html'>perMarkerQC</a></span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>,
path2plink<span class='op'>=</span><span class='va'>path2plink</span><span class='op'>)</span>

<span class='co'>## Create new dataset of individuals and markers passing QC</span>
<span class='va'>ids_all</span> <span class='op'>&lt;-</span> <span class='fu'>cleanData</span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>, macTh<span class='op'>=</span><span class='fl'>15</span>,
verbose<span class='op'>=</span><span class='cn'>TRUE</span>, path2plink<span class='op'>=</span><span class='va'>path2plink</span>, filterAncestry<span class='op'>=</span><span class='cn'>FALSE</span>,
filterRelated<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'># Run qc on subset of samples and markers in the dataset</span>
<span class='va'>highlight_samples</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"keep_individuals"</span>,
package<span class='op'>=</span><span class='st'>"plinkQC"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>remove_individuals_file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"remove_individuals"</span>,
package<span class='op'>=</span><span class='st'>"plinkQC"</span><span class='op'>)</span>

<span class='va'>fail_individuals</span> <span class='op'>&lt;-</span> <span class='fu'><a href='perIndividualQC.html'>perIndividualQC</a></span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>,
dont.check_ancestry <span class='op'>=</span> <span class='cn'>TRUE</span>, interactive<span class='op'>=</span><span class='cn'>FALSE</span>, verbose<span class='op'>=</span><span class='cn'>FALSE</span>,
highlight_samples <span class='op'>=</span> <span class='va'>highlight_samples</span><span class='op'>[</span>,<span class='fl'>2</span><span class='op'>]</span>, highlight_type <span class='op'>=</span> <span class='st'>"label"</span>,
remove_individuals <span class='op'>=</span> <span class='va'>remove_individuals_file</span>, path2plink<span class='op'>=</span><span class='va'>path2plink</span><span class='op'>)</span>

<span class='co'>## Run marker QC checks</span>
<span class='va'>fail_markers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='perMarkerQC.html'>perMarkerQC</a></span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>,
path2plink<span class='op'>=</span><span class='va'>path2plink</span><span class='op'>)</span>

<span class='co'>## Create new dataset of individuals and markers passing QC</span>
<span class='va'>ids_all</span> <span class='op'>&lt;-</span> <span class='fu'>cleanData</span><span class='op'>(</span>indir<span class='op'>=</span><span class='va'>indir</span>, qcdir<span class='op'>=</span><span class='va'>qcdir</span>, name<span class='op'>=</span><span class='va'>name</span>, macTh<span class='op'>=</span><span class='fl'>15</span>,
verbose<span class='op'>=</span><span class='cn'>TRUE</span>, path2plink<span class='op'>=</span><span class='va'>path2plink</span>, filterAncestry<span class='op'>=</span><span class='cn'>FALSE</span>,
remove_individuals <span class='op'>=</span> <span class='va'>remove_individuals_file</span><span class='op'>)</span>
<span class='op'>}</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


