<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Quality control for all markers in plink-dataset — perMarkerQC • plinkQC</title>

<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Quality control for all markers in plink-dataset — perMarkerQC" />

<meta property="og:description" content="perMarkerQC checks the markers in the plink dataset for their missingness
rates across samples, their deviation from Hardy-Weinberg-Equilibrium (HWE)
and their minor allele frequencies (MAF). Per default, it assumes that IDs of
individuals that have failed perIndividualQC have been written
to qcdir/alg.failIDs and removes these individuals when computing missingness
rates, HWE p-values and MAF. If the qcdir/alg.failIDs file does not exist, a
message is written to stdout but the analyses will continue for all samples
in the alg.fam/al.bed/alg.bim dataset.
Depicts i) SNP missingness rates (stratified by minor allele
frequency) as histograms, ii) p-values of HWE exact test (stratified by all
and low p-values) as histograms and iii) the minor allele frequency
distribution as a histogram." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/plinkQC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Quality control for all markers in plink-dataset</h1>
    <small class="dont-index">Source: <a href='https://github.com/HannahVMeyer/plinkQC/blob/master/R/markerQC.R'><code>R/markerQC.R</code></a></small>
    <div class="hidden name"><code>perMarkerQC.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>perMarkerQC checks the markers in the plink dataset for their missingness
rates across samples, their deviation from Hardy-Weinberg-Equilibrium (HWE)
and their minor allele frequencies (MAF). Per default, it assumes that IDs of
individuals that have failed <code><a href='perIndividualQC.html'>perIndividualQC</a></code> have been written
to qcdir/alg.failIDs and removes these individuals when computing missingness
rates, HWE p-values and MAF. If the qcdir/alg.failIDs file does not exist, a
message is written to stdout but the analyses will continue for all samples
in the alg.fam/al.bed/alg.bim dataset.
Depicts i) SNP missingness rates (stratified by minor allele
frequency) as histograms, ii) p-values of HWE exact test (stratified by all
and low p-values) as histograms and iii) the minor allele frequency
distribution as a histogram.</p>
    
    </div>

    <pre class="usage"><span class='fu'>perMarkerQC</span>(<span class='no'>qcdir</span>, <span class='no'>alg</span>, <span class='kw'>mafTh</span> <span class='kw'>=</span> <span class='fl'>0.01</span>, <span class='kw'>macTh</span> <span class='kw'>=</span> <span class='fl'>20</span>, <span class='kw'>hweTh</span> <span class='kw'>=</span> <span class='fl'>1e-05</span>,
  <span class='kw'>lmissTh</span> <span class='kw'>=</span> <span class='fl'>0.01</span>, <span class='kw'>interactive</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>path2plink</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>showPlinkOutput</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>do.check_snp_missingness</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>do.check_hwe</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>do.check_maf</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>qcdir</th>
      <td><p>[character] /path/to/directory/with/QC/results containing
alg.bim, alg.bed, alg.fam files and alg.fail.IDs containing IIDs of
individuals that failed QC. User needs writing permission to qcdir.</p></td>
    </tr>
    <tr>
      <th>alg</th>
      <td><p>[character] prefix of QC-ed plink files, i.e. alg.bed, alg.bim,
alg.fam.</p></td>
    </tr>
    <tr>
      <th>mafTh</th>
      <td><p>[double] Threshold for minor allele frequency cut-off.</p></td>
    </tr>
    <tr>
      <th>macTh</th>
      <td><p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples)</p></td>
    </tr>
    <tr>
      <th>hweTh</th>
      <td><p>[double] Significance threshold for deviation from HWE.</p></td>
    </tr>
    <tr>
      <th>lmissTh</th>
      <td><p>[double] Threshold for acceptable variant missing rate across
samples.</p></td>
    </tr>
    <tr>
      <th>interactive</th>
      <td><p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_marker) via ggplot2::ggsave(p=p_marker,
other_arguments) or pdf(outfile) print(p_marker) dev.off().</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>[logical] If TRUE, progress info is printed to standard out.</p></td>
    </tr>
    <tr>
      <th>path2plink</th>
      <td><p>[character] Absolute path to where external plink software
<a href='https://www.cog-genomics.org/plink/1.9/'>https://www.cog-genomics.org/plink/1.9/</a> can be found. If not provided,
assumed that PATH set-up works and plink will be found by system("plink").</p></td>
    </tr>
    <tr>
      <th>showPlinkOutput</th>
      <td><p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p></td>
    </tr>
    <tr>
      <th>do.check_snp_missingness</th>
      <td><p>[logical] if TRUE, run
<code><a href='check_snp_missingness.html'>check_snp_missingness</a></code>.</p></td>
    </tr>
    <tr>
      <th>do.check_hwe</th>
      <td><p>[logical] if TRUE, run <code><a href='check_hwe.html'>check_hwe</a></code></p></td>
    </tr>
    <tr>
      <th>do.check_maf</th>
      <td><p>[logical] if TRUE, run <code><a href='check_maf.html'>check_maf</a></code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>named [list] with i) fail_list, a named [list] with 1.
SNP_missingness, containing SNP IDs failing the missingness threshold
lmissTh, 2. hwe, containing SNP IDs failing the HWE exact test threshold
hweTh and 3. maf, containing SNPs failing the MAF threshold mafTh/MAC
threshold macTh and ii) p_markerQC, a ggplot2-object 'containing' a
sub-paneled plot with the QC-plots of <code><a href='check_snp_missingness.html'>check_snp_missingness</a></code>,
<code><a href='check_hwe.html'>check_hwe</a></code> and <code><a href='check_maf.html'>check_maf</a></code>, which can
be shown by print(p_markerQC).
List entries contain NULL if that specific check was not chosen.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>perMarkerQC wraps around the marker QC functions
<code><a href='check_snp_missingness.html'>check_snp_missingness</a></code>, <code><a href='check_hwe.html'>check_hwe</a></code> and
<code><a href='check_maf.html'>check_maf</a></code>. For details on the parameters and outputs, check
these function documentations.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

