<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run PLINK IBD estimation — run_check_relatedness • plinkQC</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Run PLINK IBD estimation — run_check_relatedness" />
<meta property="og:description" content="Run LD pruning on dataset with plink --exclude range highldfile
--indep-pairwise 50 5 0.2, where highldfile contains regions of high LD as
provided by Anderson et (2010) Nature Protocols. Subsequently, plink
--genome is run on the LD pruned, maf-filtered data. plink --genome
calculates identity by state (IBS) for each pair of individuals based on the
average proportion of alleles shared at genotyped SNPs. The degree of recent
shared ancestry,i.e. the identity by descent (IBD) can be estimated from the
genome-wide IBS. The proportion of IBD between two individuals is returned by
--genome as PI_HAT." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-standard navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">plinkQC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-file"></span>
     
    documentation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/plinkQC.html">Step-by-step QC</a>
    </li>
    <li>
      <a href="../articles/AncestryCheck.html">Ancestry</a>
    </li>
    <li>
      <a href="../articles/HapMap.html">HapMap III reference</a>
    </li>
    <li>
      <a href="../articles/Genomes1000.html">1000 Genomes reference</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/HannahVMeyer/plinkQC">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run PLINK IBD estimation</h1>
    <small class="dont-index">Source: <a href='https://github.com/HannahVMeyer/plinkQC/blob/master/R/individualQC.R'><code>R/individualQC.R</code></a></small>
    <div class="hidden name"><code>run_check_relatedness.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run LD pruning on dataset with plink --exclude range highldfile
--indep-pairwise 50 5 0.2, where highldfile contains regions of high LD as
provided by Anderson et (2010) Nature Protocols. Subsequently, plink
--genome is run on the LD pruned, maf-filtered data. plink --genome
calculates identity by state (IBS) for each pair of individuals based on the
average proportion of alleles shared at genotyped SNPs. The degree of recent
shared ancestry,i.e. the identity by descent (IBD) can be estimated from the
genome-wide IBS. The proportion of IBD between two individuals is returned by
--genome as PI_HAT.</p>
    </div>

    <pre class="usage"><span class='fu'>run_check_relatedness</span>(<span class='no'>indir</span>, <span class='no'>name</span>, <span class='kw'>qcdir</span> <span class='kw'>=</span> <span class='no'>indir</span>, <span class='kw'>highIBDTh</span> <span class='kw'>=</span> <span class='fl'>0.185</span>,
  <span class='kw'>mafThRelatedness</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>path2plink</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>genomebuild</span> <span class='kw'>=</span> <span class='st'>"hg19"</span>,
  <span class='kw'>showPlinkOutput</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>indir</th>
      <td><p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p></td>
    </tr>
    <tr>
      <th>qcdir</th>
      <td><p>[character] /path/to/directory to save name.genome as returned
by plink --genome. User needs writing permission to qcdir. Per default
qcdir=indir.</p></td>
    </tr>
    <tr>
      <th>highIBDTh</th>
      <td><p>[double] Threshold for acceptable proportion of IBD between
pair of individuals; only pairwise relationship estimates larger than this
threshold will be recorded.</p></td>
    </tr>
    <tr>
      <th>mafThRelatedness</th>
      <td><p>[double] Threshold of minor allele frequency filter
for selecting variants for IBD estimation.</p></td>
    </tr>
    <tr>
      <th>path2plink</th>
      <td><p>[character] Absolute path to PLINK executable
(<a href='https://www.cog-genomics.org/plink/1.9/'>https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accesible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code><a href='https://rdrr.io/pkg/sys/man/exec.html'>exec_wait</a></code>('plink').</p></td>
    </tr>
    <tr>
      <th>genomebuild</th>
      <td><p>[character] Name of the genome build of the PLINK file
annotations, ie mappings in the name.bim file. Will be used to remove
high-LD regions based on the coordinates of the respective build. Options
are hg18, hg19 and hg38. See @details.</p></td>
    </tr>
    <tr>
      <th>showPlinkOutput</th>
      <td><p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>[logical] If TRUE, progress info is printed to standard out.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Both <code>run_check_relatedness</code> and its evaluation via
<code><a href='evaluate_check_relatedness.html'>evaluate_check_relatedness</a></code> can simply be invoked by
<code><a href='check_relatedness.html'>check_relatedness</a></code>.</p>
<p>The IBD estimation is conducted on LD pruned data and in a first
step, high LD regions are excluded. The regions were derived from the
high-LD-regions file provided by Anderson et (2010) Nature Protocols. These
regions are in NCBI36 (hg18) coordinates and were lifted to GRCh37 (hg19)
and GRC38 (hg38) coordinates using the liftOver tool available here:
<a href='https://genome.ucsc.edu/cgi-bin/hgLiftOver'>https://genome.ucsc.edu/cgi-bin/hgLiftOver</a>. The 'Minimum ratio of bases
that must remap' which was set to 0.5 and the 'Allow multiple output regions'
box ticked; for all other parameters, the default options were selected.
LiftOver files were generated on July 9,2019. The commands for formatting
the files are provided in system.file("extdata", 'liftOver.cmd',
package="plinkQC").</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>indir</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"plinkQC"</span>)
<span class='no'>name</span> <span class='kw'>&lt;-</span> <span class='st'>'data'</span>
<span class='no'>qcdir</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span>()
<span class='co'># the following code is not run on package build, as the path2plink on the</span>
<span class='co'># user system is not known.</span></div><span class='co'># NOT RUN {</span>
<span class='no'>run</span> <span class='kw'>&lt;-</span> <span class='fu'>run_check_relatedness</span>(<span class='kw'>indir</span><span class='kw'>=</span><span class='no'>indir</span>, <span class='kw'>qcdir</span><span class='kw'>=</span><span class='no'>qcdir</span>, <span class='kw'>name</span><span class='kw'>=</span><span class='no'>name</span>)
<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Hannah Meyer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9100.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


